---
title: "IFNG8_NGG"
author: "Matt Coelho"
date: "15/12/2021"
output: html_document
---

```{r}
# author: Matt Coelho
# Copyright (C) <2021> Genome Research Ltd.
# See License - MIT License
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# load packages
library(tidyverse)
library(reshape2)
library(ggrepel)
library(stringr)
```

```{r}
# read in guides for JAK1 from WGE
data <- as_tibble(read.csv("bedit_jak1_sgrnas_v1.1_annotated_library_NG.csv", stringsAsFactors = FALSE, header = TRUE))
```

```{r}
# determine window for base editor (here is 4-9). This is lenient and includes guides with G mismatch at pos 1.
data <-  mutate(data, window = ifelse(strand == "+", substr(seq, 4, 9), substr(seq, 15, 20)))

# edit window (here A to G for ABE)
data <-  mutate(data, edit = ifelse(strand == "+", str_replace_all(window, "A", "g"), str_replace_all(window, "T", "c")))

# make mutant seq (optional making full mutant protospacer genotype)
data <-  mutate(data, mutant = ifelse(strand == "+", paste0(substr(seq, 1, 3), edit, substr(seq, 10, 23)), paste0(substr(seq, 1, 14), edit, substr(seq, 21, 23))))

```

```{r}
# give numbers of each gRNA_type in the library
data %>% count(sgRNA_type)
data %>% distinct() %>% nrow()
```

```{r}
# add a FORWARD guide column (reverse reverse complement for - strand seq)
# add 5' G for non-G starting guides. You can use this to ORDER GUIDE libraries (add appropriate Gibson cloning adapters)
# get rid of PAM
data <- mutate(data, guide = ifelse(strand == "+", substr(seq, 1, 20), substr(seq, 4, 23)))

# complement
data <- mutate(data, guide = ifelse(strand == "+", guide, str_replace_all(guide, "A", "t")))
data <- mutate(data, guide = ifelse(strand == "+", guide, str_replace_all(guide, "T", "a")))
data <- mutate(data, guide = ifelse(strand == "+", guide, str_replace_all(guide, "C", "g")))
data <- mutate(data, guide = ifelse(strand == "+", guide, str_replace_all(guide, "G", "c")))
data <- mutate(data, guide = str_to_upper(guide))

# reverse
data <- mutate(data, guide = ifelse(strand == "+", guide, paste0(substr(guide, 20, 20), substr(guide, 19, 19),substr(guide, 18, 18),substr(guide, 17, 17),substr(guide, 16, 16),substr(guide, 15, 15), substr(guide, 14, 14), substr(guide, 13, 13), substr(guide, 12, 12), substr(guide, 11, 11), substr(guide, 10, 10), substr(guide, 9, 9), substr(guide, 8, 8), substr(guide, 7, 7), substr(guide, 6, 6), substr(guide, 5, 5), substr(guide, 4, 4), substr(guide, 3, 3), substr(guide, 2, 2), substr(guide, 1, 1))))

# add a G for the expression from U6 promtoter by replaceing the base at pos. 1 if, it is not a G
data <- mutate(data, G_guide = ifelse(substr(guide, 1, 1) == "G", guide, paste0("G", substr(guide, 2, 20))))

# primers
# forward_primer
data <- mutate(data, for_primer = paste0("CACC", G_guide))

# reverse_primer
data <- mutate(data, rev_primer = G_guide)
data <- mutate(data, rev_primer = str_replace_all(rev_primer, "A", "t"))
data <- mutate(data, rev_primer = str_replace_all(rev_primer, "T", "a"))
data <- mutate(data, rev_primer = str_replace_all(rev_primer, "C", "g"))
data <- mutate(data, rev_primer = str_replace_all(rev_primer, "G", "c"))
data <- mutate(data, rev_primer = str_to_upper(rev_primer))

data <- mutate(data, rev_primer = paste0(substr(rev_primer, 20, 20), substr(rev_primer, 19, 19),substr(rev_primer, 18, 18),substr(rev_primer, 17, 17),substr(rev_primer, 16, 16),substr(rev_primer, 15, 15), substr(rev_primer, 14, 14), substr(rev_primer, 13, 13), substr(rev_primer, 12, 12), substr(rev_primer, 11, 11), substr(rev_primer, 10, 10), substr(rev_primer, 9, 9), substr(rev_primer, 8, 8), substr(rev_primer, 7, 7), substr(rev_primer, 6, 6), substr(rev_primer, 5, 5), substr(rev_primer, 4, 4), substr(rev_primer, 3, 3), substr(rev_primer, 2, 2), substr(rev_primer, 1, 1)))

data <- mutate(data, rev_primer = paste0("AAAC", rev_primer))
data %>% count(sgRNA_type)
```

```{r}
#read in gRNA off-target reports
off_targets_NGN <-  as_tibble(read.csv("off_targets_NGN_211221.csv", stringsAsFactors = FALSE, header = TRUE))
off_targets_NGN <- mutate(off_targets_NGN, sgRNA_ID = as.character(sgRNA_ID))

off_targets_NGG <-  as_tibble(read.csv("off_targets_NGG_311221.csv", stringsAsFactors = FALSE, header = TRUE))
off_targets_NGG <- mutate(off_targets_NGG, sgRNA_ID = as.character(sgRNA_ID))

data <- left_join(data, off_targets_NGN)
data <- left_join(data, off_targets_NGG)

data <- mutate(data, Gene = ifelse(sgRNA_type == "intergenic" | sgRNA_type == "non_targeting", "N/A", Gene))
data %>% count(sgRNA_type)
```

```{r}
# read in zscores for JAK1 NGG screens
zscores_NGG <- as_tibble(read.csv("160321_JAK1_zscores_NGG.csv", stringsAsFactors = FALSE, header = TRUE))
zscores_NGG <- mutate(zscores_NGG, Gene = ifelse(Gene == "NOGENE", "N/A",
  ifelse(Gene == "", "N/A", 
         Gene)))
data <- left_join(data, zscores_NGG)
data
```

```{r}
# read in zscores for JAK1 NGN BE4 CBE screens
zscores_NGN_BE4 <- as_tibble(read.csv("zscores_NG_BE4_270821.csv", stringsAsFactors = FALSE, header = TRUE))
zscores_NGN_BE4 <- mutate(zscores_NGN_BE4, Gene = ifelse(Gene == "NOGENE", "N/A",
  ifelse(Gene == "", "N/A", 
         Gene)))
data <- left_join(data, zscores_NGN_BE4)
data
```

```{r}
# read in zscores for JAK1 NGN  CBE screens
zscores_NGN <- as_tibble(read.csv("110122_zscores_2750.csv", stringsAsFactors = FALSE, header = TRUE))
zscores_NGN <- mutate(zscores_NGN, Gene = ifelse(Gene == "NOGENE", "N/A",
  ifelse(Gene == "", "N/A", 
         Gene)))
data <- left_join(data, zscores_NGN)
data
```

```{r}
# read in raw gRNA reads counts
raw_reads <- as_tibble(read.csv("raw_read_counts_161221.csv", stringsAsFactors = FALSE, header = TRUE))
raw_reads <- mutate(raw_reads, sgRNA_ID = as.character(sgRNA_ID))

# compute RPM normalised values
RPM <- function(x) ((x/sum(x) *1000000) + 1)
average <- function(x,y) ((x + y) /2)
L2FC <- function(x,y) (log2(x/y))
raw_reads <- mutate_at(raw_reads, vars(matches("read_count")), list(RPM = RPM))

raw_reads
write.csv(raw_reads, "Supplementary_Table_3info.csv")
                    
# compute average normalised gRNA
raw_reads <- mutate(raw_reads, average_iBE3_9NGN_T0_RPM = average(iBE3_9NGN_exp1_T0.read_count_RPM, iBE3_9NGN_exp2_T0.read_count_RPM))
raw_reads <- mutate(raw_reads, average_iBE3_9NGN_cntrl_RPM = average(iBE3_9NGN_exp1_cntrl.read_count_RPM, iBE3_9NGN_exp2_cntrl.read_count_RPM))
raw_reads <- mutate(raw_reads, average_iBE3_9NGN_ifng_RPM = average(iBE3_9NGN_exp1_ifng.read_count_RPM, iBE3_9NGN_exp2_ifng.read_count_RPM))

raw_reads <- mutate(raw_reads, average_iBE3_9NGN_cl12_T0_RPM = average(iBE3_9NGN_cl12_exp1_T0.read_count_RPM, iBE3_9NGN_cl12_exp2_T0.read_count_RPM))
raw_reads <- mutate(raw_reads, average_iBE3_9NGN_cl12_cntrl_RPM = average(iBE3_9NGN_cl12_exp1_cntrl.read_count_RPM, iBE3_9NGN_cl12_exp2_cntrl.read_count_RPM))
raw_reads <- mutate(raw_reads, average_iBE3_9NGN_cl12_ifng_RPM = average(iBE3_9NGN_cl12_exp1_ifng.read_count_RPM, iBE3_9NGN_cl12_exp2_ifng.read_count_RPM))

raw_reads <- mutate(raw_reads, average_iABE8eNGN_T0_RPM = average(iABE8eNGN_exp1_T0.read_count_RPM, iABE8eNGN_exp2_T0.read_count_RPM))
raw_reads <- mutate(raw_reads, average_iABE8eNGN_cntrl_RPM = average(iABE8eNGN_exp1_cntrl.read_count_RPM, iABE8eNGN_exp2_cntrl.read_count_RPM))
raw_reads <- mutate(raw_reads, average_iABE8eNGN_ifng_RPM = average(iABE8eNGN_exp1_ifng.read_count_RPM, iABE8eNGN_exp2_ifng.read_count_RPM))

raw_reads <- mutate(raw_reads, average_iABE8eNGN_cl12_T0_RPM = average(iABE8eNGN_cl12_exp1_T0.read_count_RPM, iABE8eNGN_cl12_exp2_T0.read_count_RPM))
raw_reads <- mutate(raw_reads, average_iABE8eNGN_cl12_cntrl_RPM = average(iABE8eNGN_cl12_exp1_cntrl.read_count_RPM, iABE8eNGN_cl12_exp2_cntrl.read_count_RPM))
raw_reads <- mutate(raw_reads, average_iABE8eNGN_cl12_ifng_RPM = average(iABE8eNGN_cl12_exp1_ifng.read_count_RPM, iABE8eNGN_cl12_exp2_ifng.read_count_RPM))


# compute L2FC normalised gRNA single experiments
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_Control_T0_1 = L2FC(iBE3_9NGN_exp1_cntrl.read_count_RPM, iBE3_9NGN_exp1_T0.read_count_RPM))
raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_Control_T0_2 = L2FC(iBE3_9NGN_exp2_cntrl.read_count_RPM, iBE3_9NGN_exp2_T0.read_count_RPM))

raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_cl12_Control_T0_1 = L2FC(iBE3_9NGN_cl12_exp1_cntrl.read_count_RPM, iBE3_9NGN_cl12_exp1_T0.read_count_RPM))
raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_cl12_Control_T0_2 = L2FC(iBE3_9NGN_cl12_exp2_cntrl.read_count_RPM, iBE3_9NGN_cl12_exp2_T0.read_count_RPM))

raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_IFNg_Control_1 = L2FC(iBE3_9NGN_exp1_ifng.read_count_RPM, iBE3_9NGN_exp1_cntrl.read_count_RPM))
raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_IFNg_Control_2 = L2FC(iBE3_9NGN_exp2_ifng.read_count_RPM, iBE3_9NGN_exp2_cntrl.read_count_RPM))

raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_cl12_IFNg_Control_1 = L2FC(iBE3_9NGN_cl12_exp1_ifng.read_count_RPM, iBE3_9NGN_cl12_exp1_cntrl.read_count_RPM))
raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_cl12_IFNg_Control_2 = L2FC(iBE3_9NGN_cl12_exp2_ifng.read_count_RPM, iBE3_9NGN_cl12_exp2_cntrl.read_count_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_Control_T0_1 = L2FC(iABE8eNGN_exp1_cntrl.read_count_RPM, iABE8eNGN_exp1_T0.read_count_RPM))
raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_Control_T0_2 = L2FC(iABE8eNGN_exp2_cntrl.read_count_RPM, iABE8eNGN_exp2_T0.read_count_RPM))

raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_cl12_Control_T0_1 = L2FC(iABE8eNGN_cl12_exp1_cntrl.read_count_RPM, iABE8eNGN_cl12_exp1_T0.read_count_RPM))
raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_cl12_Control_T0_2 = L2FC(iABE8eNGN_cl12_exp2_cntrl.read_count_RPM, iABE8eNGN_cl12_exp2_T0.read_count_RPM))

raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_IFNg_Control_1 = L2FC(iABE8eNGN_exp1_ifng.read_count_RPM, iABE8eNGN_exp1_cntrl.read_count_RPM))
raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_IFNg_Control_2 = L2FC(iABE8eNGN_exp2_ifng.read_count_RPM, iABE8eNGN_exp2_cntrl.read_count_RPM))

raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_cl12_IFNg_Control_1 = L2FC(iABE8eNGN_cl12_exp1_ifng.read_count_RPM, iABE8eNGN_cl12_exp1_cntrl.read_count_RPM))
raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_cl12_IFNg_Control_2 = L2FC(iABE8eNGN_cl12_exp2_ifng.read_count_RPM, iABE8eNGN_cl12_exp2_cntrl.read_count_RPM))


# compute L2FC normalised gRNA averages
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_Control_T0 = L2FC(average_iBE3_9NGN_cntrl_RPM, average_iBE3_9NGN_T0_RPM))
raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_IFNg_Control = L2FC(average_iBE3_9NGN_ifng_RPM, average_iBE3_9NGN_cntrl_RPM))

raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_cl12_Control_T0 = L2FC(average_iBE3_9NGN_cl12_cntrl_RPM, average_iBE3_9NGN_cl12_T0_RPM))
raw_reads <- mutate(raw_reads, L2FC_iBE3_9NGN_cl12_IFNg_Control = L2FC(average_iBE3_9NGN_cl12_ifng_RPM, average_iBE3_9NGN_cl12_cntrl_RPM))

#ABE NGN
raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_Control_T0 = L2FC(average_iABE8eNGN_cntrl_RPM, average_iABE8eNGN_T0_RPM))
raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_IFNg_Control = L2FC(average_iABE8eNGN_ifng_RPM, average_iABE8eNGN_cntrl_RPM))

raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_cl12_Control_T0 = L2FC(average_iABE8eNGN_cl12_cntrl_RPM, average_iABE8eNGN_cl12_T0_RPM))
raw_reads <- mutate(raw_reads, L2FC_iABE8eNGN_cl12_IFNg_Control = L2FC(average_iABE8eNGN_cl12_ifng_RPM, average_iABE8eNGN_cl12_cntrl_RPM))

# join
data <- left_join(data, raw_reads)

# filter out smaples with < 100 reads

data %>% 
  filter(iBE3_9NGN_cl12_exp1_T0.read_count <100 | iBE3_9NGN_cl12_exp2_T0.read_count <100 | iABE8eNGN_cl12_exp1_T0.read_count <100 | iABE8eNGN_cl12_exp2_T0.read_count <100 | iBE3_9NGN_cl12_exp1_cntrl.read_count <100 | iBE3_9NGN_cl12_exp2_cntrl.read_count <100 | iABE8eNGN_cl12_exp1_cntrl.read_count <100 | iABE8eNGN_cl12_exp2_cntrl.read_count <100 | iBE3_9NGN_cl12_exp1_ifng.read_count <100 | iBE3_9NGN_cl12_exp2_ifng.read_count <100 | iABE8eNGN_cl12_exp1_ifng.read_count <100 | iABE8eNGN_cl12_exp2_ifng.read_count <100) %>% 
  distinct()

data <- data %>% 
  filter(iBE3_9NGN_cl12_exp1_T0.read_count >100 & iBE3_9NGN_cl12_exp2_T0.read_count >100 & iABE8eNGN_cl12_exp1_T0.read_count >100 & iABE8eNGN_cl12_exp2_T0.read_count >100 & iBE3_9NGN_cl12_exp1_cntrl.read_count >100 & iBE3_9NGN_cl12_exp2_cntrl.read_count >100 & iABE8eNGN_cl12_exp1_cntrl.read_count >100 & iABE8eNGN_cl12_exp2_cntrl.read_count >100 & iBE3_9NGN_cl12_exp1_ifng.read_count >100 & iBE3_9NGN_cl12_exp2_ifng.read_count >100 & iABE8eNGN_cl12_exp1_ifng.read_count >100 & iABE8eNGN_cl12_exp2_ifng.read_count >100)

data
# filter out smaples with > 50000 control read outliers (1 gRNA)

data %>% 
  filter(iABE8eNGN_cl12_exp1_cntrl.read_count >50000 | iABE8eNGN_cl12_exp2_cntrl.read_count >50000 | iBE3_9NGN_cl12_exp1_cntrl.read_count > 50000 | iBE3_9NGN_cl12_exp2_cntrl.read_count > 50000) %>%
  distinct()

data <- data %>% 
  filter(iABE8eNGN_cl12_exp1_cntrl.read_count <50000 & iABE8eNGN_cl12_exp2_cntrl.read_count <50000 & iBE3_9NGN_cl12_exp1_cntrl.read_count < 50000 & iBE3_9NGN_cl12_exp2_cntrl.read_count < 50000) %>%
  distinct()

data %>% count(sgRNA_type)
data
```
```{r}
# check correlation between L2FC and zscores

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data, Gene %in% c("JAK1")), aes(x=zscore_iBE3_9NGN_cl12_ifng_vs_cntrl, y= L2FC_iBE3_9NGN_cl12_IFNg_Control)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all pathway gRNAs", x = "zscore_iBE3_9NGN_cl12_ifng_vs_cntrl", y = "L2FC_iBE3_9NGN_cl12_IFNg_Control") +
geom_smooth(method = lm) +
theme_classic()
plot1

plot2 <- ggplot(subset(data, Gene %in% c("JAK1")), aes(x=zscore_iABE8eNGN_cl12_ifng_vs_cntrl, y= L2FC_iABE8eNGN_cl12_IFNg_Control)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all pathway gRNAs", x = "zscore_iABE8eNGN_cl12_ifng_vs_cntrl_foldchange", y = "L2FC_iABE8e_NGN_IFNg_Control") +
geom_smooth(method = lm) +
theme_classic()
plot2

```

```{r}
# quality control plots
plot1 <- ggplot(data, aes(y= iBE3_9NGN_exp1_T0.read_count)) + 
geom_histogram() +
labs(title= "Reads T0 BE3.9maxNGN exp 1")
plot1

plot2 <- ggplot(data, aes(y= iBE3_9NGN_exp2_T0.read_count)) + 
geom_histogram() +
labs(title= "Reads T0 BE3.9maxNGN exp 2")
plot2

plot3 <- ggplot(data, aes(y= iABE8eNGN_exp1_T0.read_count)) + 
geom_histogram() +
labs(title= "Reads T0 ABE8e NGN exp 1")
plot3

plot4 <- ggplot(data, aes(y= iABE8eNGN_exp2_T0.read_count)) + 
geom_histogram() +
labs(title= "Reads T0 ABE8e NGN exp 2")
plot4

plot5 <- ggplot(data, aes(x= zscore_iBE3_9NGN_cl12_cntrl_vs_T0, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_iBE3_9NGN_cl12_cntrl_vs_T0") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x=  zscore_iABE8eNGN_cl12_cntrl_vs_T0, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "score_iABE8eNGN_cl12_cntrl_vs_T0") +
theme_classic()
plot6
```

```{r}
# output zscores scores
data %>% select(sgRNA_ID, zscore_iABE8eNGN_cl12_ifng_vs_cntrl, zscore_iBE3_9NGN_cl12_ifng_vs_cntrl, Gene) %>%
  write.csv("coverage_scores.csv")
```

```{r}
# find positions of edits within the window
data <-  mutate(data, VEP = ifelse(strand == "+", str_locate_all(mutant, pattern = "g"), str_locate_all(mutant, pattern = "c" )))

# index the edits and clean up the syntax for numeric input only
data <-  separate(data, VEP, remove = FALSE, into = c("first", "second", "third", "fourth", "fifth", "sixth", "extra"), sep = c("\\,"), extra = "merge")
data <-  mutate(data, first =(str_replace(first, "integer", "")))
data <-  mutate(data, first =(str_replace(first, "c", "")))
data <-  mutate(data, first =(str_replace(first, "\\(", "")))
data <-  mutate(data, first =(str_replace(first, "\\)", "")))
data <-  mutate(data, second =(str_replace(second, "\\)", "")))
data <-  mutate(data, third =(str_replace(third, "\\)", "")))
data <-  mutate(data, fourth =(str_replace(fourth, "\\)", "")))
data <-  mutate(data, fifth =(str_replace(fifth, "\\)", "")))
data <-  mutate(data, sixth =(str_replace(sixth, "\\)", "")))
```

```{r}
# make positions numeric to determine location from start
data <-  mutate(data, first = as.numeric(first))
data <-  mutate(data, second = as.numeric(second))
data <-  mutate(data, third = as.numeric(third))
data <-  mutate(data, fourth = as.numeric(fourth))
data <-  mutate(data, fifth = as.numeric(fifth))
data <-  mutate(data, sixth = as.numeric(sixth))
data <-  mutate(data, start = as.numeric(start))

# generate single VEP id window mutants 
data <- mutate(data, mut1 = ifelse(strand == "+", paste(chr, (start + (first-1)), (start + (first-1)), "A/G", strand), paste(chr, (start+(first-1)), (start+(first-1)), "T/C", "+")))
data <- mutate(data, mut1 = ifelse(first == "0" | first == "NA", "NA", mut1))

data <- mutate(data, mut2 = ifelse(strand == "+", paste(chr, (start + (second-1)), (start + (second-1)), "A/G", strand), paste(chr, (start+(second-1)), (start+(second-1)), "T/C", "+")))
data <- mutate(data, mut2 = ifelse(second == "NA" | second == first, "NA", mut2))

data <- mutate(data, mut3 = ifelse(strand == "+", paste(chr, (start + (third-1)), (start + (third-1)), "A/G", strand), paste(chr, (start+ (third-1)), (start+ (third-1)), "T/C", "+")))
data <- mutate(data, mut3 = ifelse(third == "NA" | third == first | third == second, "NA", mut3))

data <- mutate(data, mut4 = ifelse(strand == "+", paste(chr, (start + (fourth-1)), (start + (fourth-1)), "A/G", strand), paste(chr, (start+ (fourth-1)), (start+ (fourth-1)), "T/C", "+")))
data <- mutate(data, mut4 = ifelse(fourth == "NA" | fourth == first | fourth == second | fourth == third, "NA", mut4))

data <- mutate(data, mut5 = ifelse(strand == "+", paste(chr, (start + (fifth-1)), (start + (fifth-1)), "A/G", strand), paste(chr, (start+ (fifth-1)), (start + (fifth-1)), "T/C", "+")))
data <- mutate(data, mut5 = ifelse(fifth == "NA" | fifth == first | fifth == second | fifth == third | fifth == fourth, "NA", mut5))

data <- mutate(data, mut6 = ifelse(strand == "+", paste(chr, (start + (sixth-1)), (start + (sixth-1)), "A/G", strand), paste(chr, (start+ (sixth-1)), (start + (sixth-1)), "T/C", "+")))
data <- mutate(data, mut6 = ifelse(sixth == "NA" | sixth == first | sixth == second | sixth == third | sixth == fourth | sixth == fifth, "NA", mut6))

# generate combined VEP id window mutants
data <- mutate(data, combined_mut = ifelse(window != edit & strand == "+", paste0(chr, " ", (start+3), " ",(start+8), " ", window, "/", str_to_upper(edit), " ", "+"), "NA"))
data <- mutate(data, combined_mut = ifelse(window != edit & strand == "-", paste0(chr, " ", (start+14), " ",(start+19), " ", window, "/", str_to_upper(edit), " ", "+"), combined_mut))

# melt data and eliminate NA in VEP_id
data <- melt(select(data, !c(VEP, extra)), measure.vars = c("mut1", "mut2", "mut3", "mut4", "mut5", "mut6", "combined_mut"), variable.name = "edit_index", value.name = "VEP_id")
data <- mutate(data, VEP_id = str_replace_all(string = VEP_id, pattern  = "NA", replacement = ""))
data <- mutate(data, VEP_id = str_replace_na(string = VEP_id, replacement = ""))

# generate tsv output file
data %>% filter(Gene %in% c("JAK1")) %>% arrange(start) %>% select("VEP_id") %>% write_tsv("VEP_query_ABE.tsv")

# Pass VEP_query.tsv to VEP ensembl and include only MANE results using online filter
# Make a results file called VEP_results.csv
# Change excel cells to numbers as it gives dates automatically which messes things up
# Remove duplicate values for VEP_id entries before joining!
# Get rid of unwanted data output columns in excel
# you can replace commas in text editor before making the csv as this can include additional delimited columns in excel
```

```{r}
# read in VEP results file
VEP_results <- as_tibble(read.csv("VEP_results_ABE.csv", stringsAsFactors = FALSE, header = TRUE))
VEP_results <- rename(VEP_results, Uploaded_variation = "X.Uploaded_variation")

data <- separate(data, VEP_id, remove = TRUE, into = c("chr_edit", "start_edit", "end_edit", "edit_genotype", "extra_"), sep = c(" "), extra = "merge")
```

```{r}
# make joining by query result compatible
data <- mutate(data, Uploaded_variation = (paste0(chr_edit, "_", start_edit, "_", edit_genotype)))
VEP_results <- rename(VEP_results, Gene_ID_VEP = Gene)

# join VEP results to gRNA table
data <- left_join(data, VEP_results)
data <- data %>% select(!extra_)
data <- data %>% select(!"first" & !"second" & !"third" & !"fourth" & !"fifth" & !"sixth")
```

```{r}
# split HGVSp output from VEP to give Amino_Acid_Position
data <- separate(data, HGVSp, remove = FALSE, into = c("HGVSp1", "HGVSp2"), sep = c(":"), extra = "merge")

data <- mutate(data, Amino_Acid_Position = substr(HGVSp2, 6, 50))
data <- mutate(data, Amino_Acid_Position = ifelse(Consequence == "synonymous_variant", str_extract(Amino_Acid_Position, "[[:digit:]]+"), Amino_Acid_Position))
data <- mutate(data, Amino_Acid_Position = ifelse(str_detect(Amino_Acid_Position, "\\?"), "1", Amino_Acid_Position))
data <- mutate(data, Amino_Acid_Position = ifelse(str_detect(Amino_Acid_Position, "_"), "multiple edit", Amino_Acid_Position))
data <- mutate(data, Amino_Acid_Position = ifelse(Consequence != "start_lost" & Amino_Acid_Position != "multiple edit" & Consequence != "synonymous_variant", str_sub(Amino_Acid_Position, end = -4), Amino_Acid_Position))

# alter names 
data <- rename(data, Amino_Acid_Change = HGVSp2)
#data <- mutate(data, Amino_Acid_Change = ifelse(str_detect(Amino_Acid_Change, "%"), "=", Amino_Acid_Change))
data %>% select(Amino_Acid_Change, Amino_Acid_Position, Consequence) %>% filter(Consequence == "synonymous_variant")
```

```{r}
# define codon change
substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

data <- mutate(data, codon  = str_sub(Amino_Acid_Change,3,5))
data <- mutate(data, new_codon  = substrRight(Amino_Acid_Change, 3))
data <- mutate(data, new_codon = ifelse(new_codon == "%3D", codon, new_codon))
data <- mutate(data, Codon_Change = paste0(codon, ">", new_codon)) 
```

```{r}
# read in Master_JAK1_Protein_Mutations_Humans
Mutations <- as_tibble(read.csv("Master_clinical_dataset_IFNG8NGG_090621.csv", stringsAsFactors = FALSE, header = TRUE))
Mutations <- mutate(Mutations, Amino_Acid_Position = as.character(Amino_Acid_Position))

# make COSMIC AA.Mutations into three letter code for compatibility
Mutations <- mutate(Mutations, Clinical_Protein_Change_ThreeL = ifelse(Citation != "gnomAD_v3.1", str_replace_all(string = Clinical_Protein_Change, 
                c("A" = "Ala",
                  "R" = "Arg",
                  "N" = "Asn",
                  "D" = "Asp",
                  "C" = "Cys",
                  "G" = "Gly",
                  "E" = "Glu",
                  "Q" = "Gln",
                  "H" = "His",
                  "I" = "Ile",
                  "L" = "Leu",
                  "K" = "Lys",
                  "M" = "Met",
                  "P" = "Pro",
                  "F" = "Phe",
                  "S" = "Ser",
                  "T" = "Thr",
                  "W" = "Trp",
                  "Y" = "Tyr",
                  "V" = "Val",
                  "\\*" = "Ter")), Clinical_Protein_Change))

Mutations <- select(Mutations, !Clinical_Protein_Change)
Mutations <- rename(Mutations, Clinical_Protein_Change = Clinical_Protein_Change_ThreeL)

# clinical_outcomes.csv from clinical trial papers ovelapping with the IFNG8 or NG JAK1 VEP dataset
clinical_outcomes <- as_tibble(read.csv("clinical_outcomes.csv", stringsAsFactors = FALSE, header = TRUE))

Mutations <- left_join(Mutations, clinical_outcomes)

data <- mutate(data, Amino_Acid_Position = as.character(Amino_Acid_Position))
data <- left_join(data, Mutations)
```

```{r}
#graphical subset to reduce Amino_Acid_Position redundant duplicate plotting
data_graphing <- data %>%
  select(Amino_Acid_Position, Gene, sgRNA_ID, sgRNA_type, Consequence, chr, start, end, seq, PTM, Amino_Acid_Change, Clinical_Protein_Change, Citation, L2FC_iBE3_9NGN_cl12_IFNg_Control, L2FC_iBE3_9NGN_cl12_IFNg_Control_1, L2FC_iBE3_9NGN_cl12_IFNg_Control_2, L2FC_iABE8eNGN_cl12_IFNg_Control, L2FC_iABE8eNGN_cl12_IFNg_Control_1, L2FC_iABE8eNGN_cl12_IFNg_Control_2, zscore_IFNG_FACS_Average_NGG, zscore_IFNG_Proliferation_Average_NGG, zscore_IFNG_Proliferation_1_NGG, zscore_IFNG_Proliferation_2_NGG, zscore_IFNG_Cntrl_prolif_average_NG_BE4, zscore_IFNG_Cntrl_prolif_1_NG_BE4, zscore_IFNG_Cntrl_prolif_2_NG_BE4, Binarized_Clinical_Response, guide, PHENOTYPES, PAM_type, off_target_summary_NGN, off_target_summary_NGG, zscore_iABE8eNGN_cl12_ifng_vs_cntrl, zscore_iBE3_9NGN_cl12_ifng_vs_cntrl_exp1, zscore_iBE3_9NGN_cl12_ifng_vs_cntrl_exp2, zscore_iABE8eNGN_cl12_ifng_vs_cntrl_exp1, zscore_iABE8eNGN_cl12_ifng_vs_cntrl_exp2, zscore_iBE3_9NGN_cl12_ifng_vs_cntrl) %>%
  distinct()
```

```{r}
# Rename Consequences

data_graphing <- mutate(data_graphing, Consequence = ifelse(Consequence == "missense_variant", "missense", 
                                                                              ifelse(Consequence == "missense_variant_splice_region_variant", "splice variant", 
                                                                                     ifelse(Consequence == "splice_region_variant", "splice variant", 
                                                                                            ifelse(Consequence == "stop_retained_variant", "synonymous",
                                                                                                   ifelse(Consequence == "synonymous_variant", "synonymous", 
                                                                                                          ifelse(Consequence == "splice_region_variant_synonymous_variant", "splice variant",
                                                                                                                  ifelse(Consequence == "splice_acceptor_variant", "splice variant",
                                                                                                                         ifelse(Consequence == "splice_donor_variant", "splice variant",
                                                                                                                                ifelse(Consequence == "splice_region_variant_intron_variant", "splice variant",
                                                                                                                                       ifelse(Consequence == "splice_region_variant,intron_variant", "splice variant",
                                                                                                                                       ifelse(Consequence == "splice_donor_region_variant_intron_variant", "splice variant",
                                                                                                                                              ifelse(Consequence == "splice_polypyrimidine_tract_variant_intron_variant", "splice variant",
                                                                                                                                                     ifelse(Consequence == "splice_polypyrimidine_tract_variant_splice_region_variant_intron_variant", "splice variant",
                                                                                                                                                            ifelse(Consequence == "splice_donor_5th_base_variant_intron_variant", "splice variant",
                                                                                                                                                              ifelse(Consequence == "downstream_gene_variant", "UTR",
                                                                                                                                                                        ifelse(Consequence == "stop_gained_splice_region_variant", "stop codon",
                                                                                                                                                                               ifelse(Consequence == "stop_gained,splice_region_variant", "stop codon",
                                                                                                                                                                                ifelse(Consequence == "start_lost", "start lost",
                                                                                                                                                                                        ifelse(Consequence == "stop_gained_start_lost", "stop codon",
                                                                                                                                                                                                ifelse(Consequence == "upstream_gene_variant", "promoter",
                                                                                                                                                                                                      ifelse(Consequence == "intron_variant", "intron",
                                                                                                                                                                                                        ifelse(Consequence == "5_prime_UTR_variant", "5'UTR",
                                                                                                                                                                                                          ifelse(Consequence == "stop_gained", "stop codon", Consequence)))))))))))))))))))))))) %>% 
  distinct()
```

```{r}
# Replace Amino_Acid_Position with numerical values, or synonymous, or multiple_edit

#data_graphing <- mutate(data_graphing, Amino_Acid_Position = str_remove(Amino_Acid_Position, "synonymous"), Amino_Acid_Position)
data_graphing <- mutate(data_graphing, Amino_Acid_Position = str_remove(Amino_Acid_Position, "multiple edit"), Amino_Acid_Position)

#data_graphing <- mutate(data_graphing, Amino_Acid_Change = ifelse(Consequence == "synonymous", NA, Amino_Acid_Change))
data_graphing <- mutate(data_graphing, Amino_Acid_Change = ifelse(str_detect(Amino_Acid_Change, "%") == TRUE, str_replace(Amino_Acid_Change, "%3D", "="), Amino_Acid_Change))

data_graphing %>% group_by(sgRNA_ID) %>% select(sgRNA_ID, Consequence, Gene, Amino_Acid_Position, Amino_Acid_Change, Clinical_Protein_Change)

```

```{r}
#filter data_graphing subset to get rid of duplicate values and summarise them into one row per gRNA

Amino_Acid_Change <- data_graphing %>%
  group_by(sgRNA_ID) %>%
  filter(is.na(Amino_Acid_Change)==FALSE) %>%
  distinct(Amino_Acid_Change) %>%
  summarise(Amino_Acid_Change = paste(Amino_Acid_Change, collapse = ", "))

Clinical_Protein_Change <- data_graphing %>%
  group_by(sgRNA_ID) %>%
  filter(is.na(Clinical_Protein_Change)==FALSE) %>%
  distinct(Clinical_Protein_Change) %>%
  summarise(Clinical_Protein_Change = paste(Clinical_Protein_Change, collapse = ", "))

Citation <- data_graphing %>%
  group_by(sgRNA_ID) %>%
  filter(is.na(Citation)==FALSE) %>%
  distinct(Citation) %>%
  summarise(Citation = paste(Citation, collapse = ", "))

PTM <- data_graphing %>%
  group_by(sgRNA_ID) %>%
  filter(is.na(PTM)==FALSE) %>%
  distinct(PTM) %>%
  summarise(PTM = paste(PTM, collapse = ", "))

Consequence <- data_graphing %>%
  group_by(sgRNA_ID) %>%
  filter(is.na(Consequence)==FALSE) %>%
  distinct(Consequence) %>%
  summarise(Consequence = paste(Consequence, collapse = ", "))

Binarized_Clinical_Response <- data_graphing %>%
  group_by(sgRNA_ID) %>%
  filter(is.na(Binarized_Clinical_Response)==FALSE) %>%
  distinct(Binarized_Clinical_Response) %>%
  summarise(Binarized_Clinical_Response = paste(Binarized_Clinical_Response, collapse = ", "))

Amino_Acid_Position <- data_graphing %>%
  group_by(sgRNA_ID) %>%
  filter(is.na(Amino_Acid_Position)==FALSE) %>%
  distinct(Amino_Acid_Position) %>%
  summarise(Amino_Acid_Position = paste(Amino_Acid_Position, collapse = " "))

PHENOTYPES <- data_graphing %>%
  group_by(sgRNA_ID) %>%
  filter(is.na(PHENOTYPES)==FALSE) %>%
  distinct(PHENOTYPES) %>%
  summarise(PHENOTYPES = paste(PHENOTYPES, collapse = ", "))

data_graphing_filtered <- select(data_graphing, !c(Amino_Acid_Change, Clinical_Protein_Change, Citation, PTM, Consequence, Binarized_Clinical_Response, Amino_Acid_Position, PHENOTYPES)) 

data_graphing_filtered <- left_join(data_graphing_filtered, Amino_Acid_Change) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, Clinical_Protein_Change) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, Citation) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, PTM) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, Consequence) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, Binarized_Clinical_Response) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, Amino_Acid_Position) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, PHENOTYPES) %>% distinct()

data_graphing_filtered
```

```{r}
# Resolve multiple consequences for one amino acid position by severity

data_graphing_filtered <- mutate(data_graphing_filtered, Consequence = ifelse(str_detect(Consequence, "stop codon"), "stop codon",
                                                                                         ifelse(str_detect(Consequence, "start lost"), "start lost", 
                                                                                                           ifelse(str_detect(Consequence, "splice variant"), "splice variant",
                                                                                                                   ifelse(str_detect(Consequence, "missense"), "missense",
                                                                                                                  ifelse(str_detect(Consequence, "UTR"), "UTR",
                                                                                                                         ifelse(str_detect(Consequence, "synonymous"), "synonymous",
                                                                                                                         Consequence)))))))


data_graphing_filtered <- data_graphing_filtered %>% 
mutate(Consequence = ifelse(str_detect(Amino_Acid_Change, "Pro") & (str_detect(Amino_Acid_Change, "p.Pro")==FALSE) & Consequence == "missense", "missense proline", Consequence))

data_graphing_filtered %>% group_by(sgRNA_ID) %>% select(sgRNA_ID, Amino_Acid_Change, Consequence, Gene, Amino_Acid_Position, sgRNA_type)
```

```{r}
# Make a simplified amino acid position such that we can plot on a numerical scale
data_graphing_filtered <- mutate(data_graphing_filtered, Amino_Acid_Change = str_remove_all(Amino_Acid_Change, "p\\."))
data_graphing_filtered <- mutate(data_graphing_filtered, Clinical_Protein_Change = str_remove_all(Clinical_Protein_Change, "p\\."))

data_graphing_filtered <- mutate(data_graphing_filtered, Amino_Acid_Position_simple = Amino_Acid_Position)
data_graphing_filtered <- mutate(data_graphing_filtered, Amino_Acid_Position_simple = as.numeric(str_extract(Amino_Acid_Position_simple, "[[:digit:]]+")))
```

```{r}
# rename the values to join with other editors
data_graphing_filtered <- data_graphing_filtered %>%
  rename(Amino_Acid_Change_ABE = Amino_Acid_Change) %>%
    rename(Clinical_Protein_Change_ABE = Clinical_Protein_Change) %>%
      rename(Citation_ABE = Citation) %>%
        rename(PTM_ABE = PTM) %>%
          rename(Consequence_ABE = Consequence) %>%
            rename(Binarized_Clinical_Response_ABE = Binarized_Clinical_Response) %>%
              rename(Amino_Acid_Position_ABE = Amino_Acid_Position) %>%
                rename(PHENOTYPES_ABE = PHENOTYPES) %>%
                  rename(Amino_Acid_Position_simple_ABE = Amino_Acid_Position_simple)
```

```{r}
# read in CBE outcomes
BE3.9 <- as_tibble(read.csv("BE3.9_outcomes_311221.csv", stringsAsFactors = FALSE, header = TRUE))
BE3.9
data_graphing_filtered

data_graphing_filtered <- left_join(data_graphing_filtered, BE3.9)
data_graphing_filtered
```

```{r}
# read in CBE outcomes
BE4max <- as_tibble(read.csv("BE4max_YE1_outcomes_311221.csv", stringsAsFactors = FALSE, header = TRUE))
BE4max
data_graphing_filtered

data_graphing_filtered <- left_join(data_graphing_filtered, BE4max)
data_graphing_filtered
```

```{r}
# for labelling validation cohort on graphs
data_graphing_filtered <- mutate(data_graphing_filtered, validation_cohort = ifelse(sgRNA_ID %in% c("908523459", "908523616", "908514537", "908514536", "908509432", "908510622", "908509431", "908510623", "908508937", "908509672", "908509010", "908512440", "908510034", "908510164", "908509007", "908510033", "908510274", "908509428", "908510028", "908510170", "908510032", "908508944", "908508948", "908510162"), "validated gRNAs", "putative"))

# removing amino acid positions not covered (NA)
data_graphing_filtered <- data_graphing_filtered %>% 
#  arrange(Amino_Acid_Position) %>%
#  group_by(Gene) %>%
#   filter(!is.na(Amino_Acid_Position)) %>%
            distinct()
data_graphing_filtered
```

```{r}
#rename PHENOTYPES to simplify them
data_graphing_filtered <- data_graphing_filtered %>% 
  mutate(phenotype_CBE = ifelse(str_detect(PHENOTYPES_CBE, "leukemia"), "leukaemia", ifelse(str_detect(PHENOTYPES_CBE, "IMMUNE_DYS"), "immune dysregulation", NA)))
```

```{r}
# consequences to include sgRNA_type controls
data_graphing_filtered <- data_graphing_filtered %>% 
  mutate(Consequence_CBE = ifelse(sgRNA_type == "intergenic", "intergenic", 
                              ifelse(sgRNA_type == "stop_essential", "stop essential", 
                                     ifelse(sgRNA_type == "stop_nonessential", "stop non-essential",
                                                                                 ifelse(sgRNA_type == "non_targeting", "non-targeting",
                                                                                        Consequence_CBE)))))

data_graphing_filtered <- data_graphing_filtered %>% 
  mutate(Consequence_ABE = ifelse(sgRNA_type == "intergenic", "intergenic", 
                              ifelse(sgRNA_type == "stop_essential", "stop essential", 
                                     ifelse(sgRNA_type == "stop_nonessential", "stop non-essential",
                                                                                 ifelse(sgRNA_type == "non_targeting", "non-targeting",
                                                                                        Consequence_ABE)))))

data_graphing_filtered <- data_graphing_filtered %>% 
  mutate(Consequence_CBE_YE1 = ifelse(sgRNA_type == "intergenic", "intergenic", 
                              ifelse(sgRNA_type == "stop_essential", "stop essential", 
                                     ifelse(sgRNA_type == "stop_nonessential", "stop non-essential",
                                                                                 ifelse(sgRNA_type == "non_targeting", "non-targeting",
                                                                                        Consequence_CBE_YE1)))))
```

```{r}
# replicate correlation - stats
subset<- data_graphing_filtered %>% filter(Gene %in% c("JAK1"))

summary(lm(subset$zscore_iABE8eNGN_cl12_ifng_vs_cntrl_exp1 ~ subset$zscore_iABE8eNGN_cl12_ifng_vs_cntrl_exp2))
summary(lm(subset$zscore_iBE3_9NGN_cl12_ifng_vs_cntrl_exp1 ~ subset$zscore_iBE3_9NGN_cl12_ifng_vs_cntrl_exp2))
summary(lm(subset$zscore_IFNG_Cntrl_prolif_1_NG_BE4 ~ subset$zscore_IFNG_Cntrl_prolif_2_NG_BE4))
summary(lm(subset$zscore_IFNG_Proliferation_1_NGG ~ subset$zscore_IFNG_Proliferation_2_NGG))
```

```{r}
# zscore and L2FC correlation and cut-off graphs
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, Gene %in% c("JAK1") & is.na(Consequence_ABE)==FALSE), aes(x= zscore_iABE8eNGN_cl12_ifng_vs_cntrl_exp1, y = zscore_iABE8eNGN_cl12_ifng_vs_cntrl_exp2, shape = PAM_type, colour = Consequence_ABE)) +
scale_shape_manual(values=c(13, 8, 19, 0)) +
scale_colour_manual(values=c("grey50", "black", "springgreen3", "yellow", "orange","royalblue1", "grey75", "purple")) +
geom_point(alpha = 0.8, position = pos, size = 3) +
labs(title= "JAK1 variant positions of concern: ABE8e-NGN", x = "proliferation z-score screen 1", y = "proliferation z-score screen 2", shape = "PAM", colour = "consequence") +
geom_vline(xintercept=c(3, 3),linetype="dotted") +
geom_hline(yintercept=c(3, 3),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(zscore_iABE8eNGN_cl12_ifng_vs_cntrl_exp1 > 3 & Consequence_ABE == "missense" & zscore_iABE8eNGN_cl12_ifng_vs_cntrl_exp2 > 3 & Consequence_ABE == "missense", Amino_Acid_Position_ABE, "")), size = 4, position = pos) +
annotate("text", y = 9, x = 0, label = c("Rsq. adj. = 0.38"), parse = FALSE, size = 5) +
theme_classic()
plot1

plot2 <- ggplot(subset(data_graphing_filtered, Gene %in% c("JAK1") & is.na(Consequence_CBE)==FALSE), aes(x= zscore_iBE3_9NGN_cl12_ifng_vs_cntrl_exp1, y = zscore_iBE3_9NGN_cl12_ifng_vs_cntrl_exp2, shape = PAM_type, colour = Consequence_CBE)) +
scale_shape_manual(values=c(13, 8, 19, 0)) +
scale_colour_manual(values=c("grey50", "black", "yellow", "orange", "royalblue1", "red2", "grey75", "purple")) +
geom_point(alpha = 0.8, position = pos, size = 4) +
labs(title= "JAK1 variant positions of concern: BE3.9max-NGN", x = "proliferation z-score screen 1", y = "proliferation z-score screen 2", shape = "PAM", colour = "consequence") +
geom_vline(xintercept=c(3, 3),linetype="dotted") +
geom_hline(yintercept=c(3, 3),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(zscore_iBE3_9NGN_cl12_ifng_vs_cntrl_exp1 > 3 & Consequence_CBE == "missense" & zscore_iBE3_9NGN_cl12_ifng_vs_cntrl_exp2 > 3 & Consequence_CBE == "missense", Amino_Acid_Position_CBE, "")), size = 4, position = pos) +
annotate("text", y = 9, x = 0, label = c("Rsq. adj. = 0.21"), parse = FALSE, size = 5) +
theme_classic()
plot2

plot3 <- ggplot(subset(data_graphing_filtered, Gene %in% c("JAK1") & is.na(Consequence_CBE_YE1)==FALSE), aes(x= zscore_IFNG_Cntrl_prolif_1_NG_BE4, y = zscore_IFNG_Cntrl_prolif_2_NG_BE4, shape = PAM_type, colour = Consequence_CBE_YE1)) +
scale_shape_manual(values=c(13, 8, 19, 0)) +
scale_colour_manual(values=c("grey50", "black", "yellow", "orange", "red2", "grey75", "purple")) +
geom_point(alpha = 0.8, position = pos, size = 3) +
labs(title= "JAK1 variant positions of concern: BE4max-YE1-NGN", x = "proliferation z-score screen 1", y = "proliferation z-score screen 2", shape = "PAM", colour = "consequence") +
geom_vline(xintercept=c(3, 3),linetype="dotted") +
geom_hline(yintercept=c(3, 3),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(zscore_IFNG_Cntrl_prolif_1_NG_BE4 > 3 & Consequence_CBE_YE1 == "missense" & zscore_IFNG_Cntrl_prolif_2_NG_BE4 > 3 & Consequence_CBE_YE1 == "missense", Amino_Acid_Position_CBE_YE1, "")), size = 4, position = pos) +
annotate("text", y = 9, x = 0, label = c("Rsq. adj. = 0.17"), parse = FALSE, size = 5) +
theme_classic()
plot3

plot4 <- ggplot(subset(data_graphing_filtered, Gene %in% c("JAK1") & is.na(Consequence_CBE)==FALSE), aes(x= zscore_IFNG_Proliferation_1_NGG, y = zscore_IFNG_Proliferation_2_NGG, shape = PAM_type, colour = Consequence_CBE)) +
scale_shape_manual(values=c(13, 8, 19, 0)) +
scale_colour_manual(values=c("grey50", "black", "yellow", "orange", "royalblue1", "red2", "grey75", "purple")) +
geom_point(alpha = 0.8, position = pos, size = 3) +
labs(title= "JAK1 variant positions of concern: BE3-NGG", x = "proliferation z-score screen 1", y = "proliferation z-score screen 2", shape = "PAM", colour = "consequence") +
geom_vline(xintercept=c(3, 3),linetype="dotted") +
geom_hline(yintercept=c(3, 3),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_1_NGG > 3 & Consequence_CBE == "missense" & zscore_IFNG_Proliferation_1_NGG > 3 & Consequence_CBE == "missense", Amino_Acid_Position_CBE, "")), size = 4, position = pos) +
annotate("text", y = 9, x = 0, label = c("Rsq. adj. = 0.59"), parse = FALSE, size = 5) +
theme_classic()
plot4
```

```{r}
# simplify labelling for graphs - Citation_CBE
data_graphing_filtered <- mutate(data_graphing_filtered, Citation_CBE_simple = str_replace(Citation_CBE, "COSMIC", "a"))
data_graphing_filtered <- mutate(data_graphing_filtered, Citation_CBE_simple = str_replace(Citation_CBE_simple, "TCGA", "b"))
data_graphing_filtered <- mutate(data_graphing_filtered, Citation_CBE_simple = str_replace(Citation_CBE_simple, "VanAllen_etal_Science_2015", "c"))
data_graphing_filtered <- mutate(data_graphing_filtered, Citation_CBE_simple = str_replace(Citation_CBE_simple, "Shin_etal_CancerDiscovery_2016", "d"))
data_graphing_filtered <- mutate(data_graphing_filtered, Citation_CBE_simple = str_replace(Citation_CBE_simple, "Lupardus_etal_PNAS_2014", "e"))
data_graphing_filtered <- mutate(data_graphing_filtered, Citation_CBE_simple = str_replace(Citation_CBE_simple, "gnomAD_v3.1", "f"))
data_graphing_filtered <- mutate(data_graphing_filtered, Citation_CBE_simple = str_replace(Citation_CBE_simple, "ClinVar", "g"))
```

```{r}
# dot plots _ proliferation - JAK1
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, Consequence_CBE %in% c("intron", "synonymous", "UTR", "promoter", NA) ==FALSE & Gene == "JAK1"), aes(x=Amino_Acid_Position_simple_CBE, y= zscore_IFNG_Proliferation_Average_NGG, colour = Consequence_CBE)) + 
geom_point(position = pos, shape = 16, alpha = 0.80, size =3) +
  
labs(title= "", x="amino acid position", y = "zscore_IFNG_Proliferation_Average_NGG") +
scale_colour_manual(values=c("black", "orange", "royal blue", "red")) +
geom_hline(yintercept=c(0,0),linetype="dotted") +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence == "missense" | L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence == "missense", paste(Amino_Acid_Position), "")), size = 3, position = pos) +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence == "missense" & is.na(phenotype)==FALSE | L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence == "missense" & is.na(phenotype)==FALSE, paste(phenotype, Amino_Acid_Position), "")), size = 3, position = pos) +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence_ABE == "missense" & is.na(PTM_ABE)==FALSE | L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence_ABE == "missense" & is.na(PTM_ABE)==FALSE, paste(PTM_ABE, Amino_Acid_Position_ABE), "")), size = 4, position = pos) +

geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_Average_NGG > 5 & Consequence_CBE %in% c("missense", "missense proline") & is.na(Citation_CBE)==FALSE| zscore_IFNG_Proliferation_Average_NGG < -2 & Consequence_CBE %in% c("missense", "missense proline") & is.na(Citation_CBE)==FALSE, paste(Amino_Acid_Position_CBE, Citation_CBE_simple), "")), size = 4, position = pos) +
  
annotate("rect", xmin =34, xmax = 420, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =220, y = -2.3, label = "FERM", size = 4, colour = "grey45") +
annotate("rect", xmin =439, xmax = 544, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =500, y = -2.3, label = "SH2-like", size = 4, colour = "grey45") +
annotate("rect", xmin =583, xmax = 855, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =710, y = -2.3, label = "pseudokinase", size = 4, colour = "grey45") +
annotate("rect", xmin =875, xmax = 1153, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =1000, y = -2.3, label = "kinase", size = 4, colour = "grey45") +
#ylim(-4, 4) +
theme_classic()
plot1
```

```{r}
# dot plots _ proliferation - JAK1
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, Consequence_CBE_YE1 %in% c("synonymous","UTR", "promoter", NA) ==FALSE & Gene == "JAK1"), aes(x=Amino_Acid_Position_simple_CBE_YE1, y= zscore_IFNG_Cntrl_prolif_average_NG_BE4, colour = Consequence_CBE_YE1)) + 
geom_point(position = pos, shape = 16, alpha = 0.80, size =3) +
labs(title= "", x="amino acid position", y = "zscore_IFNG_Cntrl_prolif_average_NG_BE4") +
scale_colour_manual(values=c("black", "orange", "red")) +
geom_hline(yintercept=c(0,0),linetype="dotted") +

geom_text_repel(aes(label=ifelse(zscore_IFNG_Cntrl_prolif_average_NG_BE4 > 5 & Consequence_CBE_YE1 %in% c("missense", "missense proline") & is.na(Clinical_Protein_Change_CBE_YE1)==FALSE| zscore_IFNG_Cntrl_prolif_average_NG_BE4 < -3 & Consequence_CBE_YE1 %in% c("missense", "missense proline") & is.na(Clinical_Protein_Change_CBE_YE1)==FALSE, paste0(Amino_Acid_Change_CBE_YE1, "\n", "(", Clinical_Protein_Change_CBE_YE1, ")"), "")), size = 3, position = pos) +

#geom_text_repel(aes(label=ifelse(zscore_iBE3_9NGN_cl12_ifng_vs_cntrl > 1.5 & Consequence_CBE == "missense" & is.na(phenotype_CBE)==FALSE | zscore_iBE3_9NGN_cl12_ifng_vs_cntrl < -1 & Consequence_CBE == "missense" & is.na(phenotype_CBE)==FALSE, paste(phenotype_CBE, Amino_Acid_Position_CBE), "")), size = 4, position = pos) +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence_ABE == "missense" & is.na(PTM_ABE)==FALSE | L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence_ABE == "missense" & is.na(PTM_ABE)==FALSE, paste(PTM_ABE, Amino_Acid_Position_ABE), "")), size = 4, position = pos) +

#geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_Average_NGG > 5 & Consequence_CBE == "missense" & is.na(Citation_CBE)==FALSE| zscore_IFNG_Proliferation_Average_NGG < -2 & Consequence_CBE == "missense" & is.na(Citation_CBE)==FALSE, paste(Amino_Acid_Position_CBE, Citation_CBE), "")), size = 4, position = pos) +
  
annotate("rect", xmin =34, xmax = 420, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =220, y = -2.3, label = "FERM", size = 4, colour = "grey45") +
annotate("rect", xmin =439, xmax = 544, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =500, y = -2.3, label = "SH2-like", size = 4, colour = "grey45") +
annotate("rect", xmin =583, xmax = 855, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =710, y = -2.3, label = "pseudokinase", size = 4, colour = "grey45") +
annotate("rect", xmin =875, xmax = 1153, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =1000, y = -2.3, label = "kinase", size = 4, colour = "grey45") +
#ylim(-4, 4) +
theme_classic()
plot1
```

```{r}
# simplify labelling for graphs - phenotype_CBE
data_graphing_filtered <- data_graphing_filtered %>%
  select(!(X))

data_graphing_filtered <- mutate(data_graphing_filtered, phenotype_CBE_simple = ifelse(phenotype_CBE == "immune dysregulation", "a",
                                                                                        ifelse(phenotype_CBE == "leukaemia", "b", phenotype_CBE)))
```

```{r}
# dot plots _ proliferation - JAK1
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, Consequence_CBE %in% c("intron", "synonymous","UTR", "promoter", NA) ==FALSE & Gene == "JAK1"), aes(x=Amino_Acid_Position_simple_CBE, y= zscore_iBE3_9NGN_cl12_ifng_vs_cntrl, colour = Consequence_CBE)) + 
geom_point(position = pos, shape = 16, alpha = 0.80, size =3) +
labs(title= "", x="amino acid position", y = "zscore_iBE3_9NGN_cl12_ifng_vs_cntrl") +
scale_colour_manual(values=c("black", "orange", "royal blue", "red")) +
geom_hline(yintercept=c(0,0),linetype="dotted") +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence == "missense" | L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence == "missense", paste(Amino_Acid_Position), "")), size = 3, position = pos) +

geom_text_repel(aes(label=ifelse(zscore_iBE3_9NGN_cl12_ifng_vs_cntrl > 1.5 & Consequence_CBE %in% c("missense", "missense proline") & is.na(phenotype_CBE)==FALSE | zscore_iBE3_9NGN_cl12_ifng_vs_cntrl < -0.8 & Consequence_CBE %in% c("missense", "missense proline") & is.na(phenotype_CBE)==FALSE, paste(Amino_Acid_Position_CBE, phenotype_CBE_simple), "")), size = 4, position = pos) +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence_ABE == "missense" & is.na(PTM_ABE)==FALSE | L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence_ABE == "missense" & is.na(PTM_ABE)==FALSE, paste(PTM_ABE, Amino_Acid_Position_ABE), "")), size = 4, position = pos) +

#geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_Average_NGG > 5 & Consequence_CBE == "missense" & is.na(Citation_CBE)==FALSE| zscore_IFNG_Proliferation_Average_NGG < -2 & Consequence_CBE == "missense" & is.na(Citation_CBE)==FALSE, paste(Amino_Acid_Position_CBE, Citation_CBE), "")), size = 3, position = pos) +
  
annotate("rect", xmin =34, xmax = 420, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =220, y = -2.3, label = "FERM", size = 4, colour = "grey45") +
annotate("rect", xmin =439, xmax = 544, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =500, y = -2.3, label = "SH2-like", size = 4, colour = "grey45") +
annotate("rect", xmin =583, xmax = 855, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =710, y = -2.3, label = "pseudokinase", size = 4, colour = "grey45") +
annotate("rect", xmin =875, xmax = 1153, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =1000, y = -2.3, label = "kinase", size = 4, colour = "grey45") +
#ylim(-4, 4) +
theme_classic()
plot1
```

```{r}
# dot plots _ proliferation - JAK1
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, Consequence_ABE %in% c("intron", "synonymous","UTR", "promoter", NA) ==FALSE & Gene == "JAK1"), aes(x=Amino_Acid_Position_simple_ABE, y= L2FC_iABE8eNGN_cl12_IFNg_Control, colour = Consequence_ABE)) + 
geom_point(position = pos, shape = 16, alpha = 0.80, size =3) +
labs(title= "", x="amino acid position", y = "L2FC: ABE8e.9 NGN control vs IFNG") +
scale_colour_manual(values=c("black", "springgreen4", "orange", "royal blue", "red")) +
geom_hline(yintercept=c(0,0),linetype="dotted") +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence == "missense" | L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence == "missense", paste(Amino_Acid_Position), "")), size = 3, position = pos) +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence == "missense" & is.na(phenotype)==FALSE | L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence == "missense" & is.na(phenotype)==FALSE, paste(phenotype, Amino_Acid_Position), "")), size = 3, position = pos) +

geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 0.75 & Consequence_ABE %in% c("missense") & is.na(PTM_ABE)==FALSE & PTM_ABE=="phos" | L2FC_iABE8eNGN_cl12_IFNg_Control < -0.7 & Consequence_ABE %in% c("missense") & is.na(PTM_ABE)==FALSE & PTM_ABE=="ub", paste(Amino_Acid_Position_ABE, PTM_ABE), "")), size = 4, position = pos) +

#geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & Consequence == "missense" & is.na(Citation)==FALSE| L2FC_iABE8eNGN_cl12_IFNg_Control < -1 & Consequence == "missense" & is.na(Citation)==FALSE, paste(Amino_Acid_Position), "")), size = 3, position = pos) +
  
annotate("rect", xmin =34, xmax = 420, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =220, y = -2.3, label = "FERM", size = 4, colour = "grey45") +
annotate("rect", xmin =439, xmax = 544, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =500, y = -2.3, label = "SH2-like", size = 4, colour = "grey45") +
annotate("rect", xmin =583, xmax = 855, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =710, y = -2.3, label = "pseudokinase", size = 4, colour = "grey45") +
annotate("rect", xmin =875, xmax = 1153, ymin = -2, ymax = 2, alpha = 0.1) +
annotate("text", x =1000, y = -2.3, label = "kinase", size = 4, colour = "grey45") +
#ylim(-4, 4) +
theme_classic()
plot1
```

```{r}
# correlation between datasets
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, Gene %in% c("JAK1")), aes(x= zscore_iBE3_9NGN_cl12_ifng_vs_cntrl, y = L2FC_iABE8eNGN_cl12_IFNg_Control)) +
geom_point(alpha = 0.5, position = pos, size = 3) +
labs(title= "correlation", x = "zscore_iBE3_9NGN_cl12_ifng_vs_cntrl", y = "L2FC_iABE8eNGN_cl12_IFNg_Control") +
stat_smooth(method = "lm") +
geom_text_repel(aes(label=ifelse(L2FC_iABE8eNGN_cl12_IFNg_Control > 1.5 & zscore_iBE3_9NGN_cl12_ifng_vs_cntrl > 1.5, paste(Amino_Acid_Position_ABE, Amino_Acid_Position_CBE), "")), size = 4, position = pos) +
theme_classic()
plot1

plot2 <- ggplot(subset(data_graphing_filtered, Gene %in% c("JAK1")), aes(x= zscore_iBE3_9NGN_cl12_ifng_vs_cntrl, y = zscore_IFNG_Cntrl_prolif_average_NG_BE4)) +
geom_point(alpha = 0.5, position = pos, size = 3) +
labs(title= "correlation", x = "zscore_iBE3_9NGN_cl12_ifng_vs_cntrl", y = "zscore_IFNG_Cntrl_prolif_average_NG_BE4") +
stat_smooth(method = "lm") +
theme_classic()
plot2

plot3 <- ggplot(subset(data_graphing_filtered, Gene %in% c("JAK1")), aes(x= zscore_iBE3_9NGN_cl12_ifng_vs_cntrl, y = zscore_IFNG_Proliferation_Average_NGG)) +
geom_point(alpha = 0.5, position = pos, size = 3) +
labs(title= "correlation", x = "zscore_iBE3_9NGN_cl12_ifng_vs_cntrl", y = "zscore_IFNG_Proliferation_Average_NGG") +
stat_smooth(method = "lm") +
theme_classic()
plot3

plot4 <- ggplot(subset(data_graphing_filtered, Gene %in% c("JAK1")), aes(x= zscore_IFNG_Proliferation_Average_NGG, y = zscore_IFNG_Cntrl_prolif_average_NG_BE4)) +
geom_point(alpha = 0.5, position = pos, size = 3) +
labs(title= "correlation", x = "zscore_IFNG_Proliferation_Average_NGG", y = "zscore_IFNG_Cntrl_prolif_average_NG_BE4") +
stat_smooth(method = "lm") +
theme_classic()
plot4
```

```{r}
# app data 
# BE3_NGG JAK1 focussed
BE3_NGG <- data_graphing_filtered %>% 
  mutate(editor = "BE3-NGG (JAK1)") %>%
  #mutate(zscore_FACS = NA) %>%
  select(Gene, guide, editor, zscore_IFNG_Proliferation_Average_NGG, zscore_IFNG_FACS_Average_NGG, Amino_Acid_Change_CBE, Amino_Acid_Position_CBE, Consequence_CBE, Clinical_Protein_Change_CBE, Citation_CBE, PTM_CBE, Amino_Acid_Position_simple_CBE, off_target_summary_NGG, off_target_summary_NGN,PHENOTYPES_CBE, sgRNA_ID) %>%
  rename(zscore_proliferation = zscore_IFNG_Proliferation_Average_NGG) %>%
  rename(zscore_FACS = zscore_IFNG_FACS_Average_NGG) %>%
  rename(Amino_Acid_Change = Amino_Acid_Change_CBE) %>%
  rename(Amino_Acid_Position = Amino_Acid_Position_CBE) %>%
  rename(Consequence = Consequence_CBE) %>%
  rename(Clinical_Protein_Change = Clinical_Protein_Change_CBE) %>%
  rename(Citation = Citation_CBE) %>%
  rename(PTM = PTM_CBE) %>%
  rename(Amino_Acid_Position_simple = Amino_Acid_Position_simple_CBE) %>%
  rename(phenotype = PHENOTYPES_CBE) %>%
  filter(Gene == "JAK1")

# BE4max_YE1_NGN
BE4max_YE1_NGN <- data_graphing_filtered %>% 
  mutate(editor = "BE4max-YE1-NGN (JAK1)") %>%
  mutate(zscore_FACS = NA) %>%
  select(Gene, guide, editor, zscore_IFNG_Cntrl_prolif_average_NG_BE4, zscore_FACS, Amino_Acid_Change_CBE_YE1, Amino_Acid_Position_CBE_YE1, Consequence_CBE_YE1, Clinical_Protein_Change_CBE_YE1, Citation_CBE_YE1, PTM_CBE_YE1, Amino_Acid_Position_simple_CBE_YE1, off_target_summary_NGG, off_target_summary_NGN, PHENOTYPES_CBE_YE1, sgRNA_ID) %>%
  rename(zscore_proliferation = zscore_IFNG_Cntrl_prolif_average_NG_BE4) %>%
  rename(Amino_Acid_Change = Amino_Acid_Change_CBE_YE1) %>%
  rename(Amino_Acid_Position = Amino_Acid_Position_CBE_YE1) %>%
  rename(Consequence = Consequence_CBE_YE1) %>%
  rename(Clinical_Protein_Change = Clinical_Protein_Change_CBE_YE1) %>%
  rename(Citation = Citation_CBE_YE1) %>%
  rename(PTM = PTM_CBE_YE1) %>%
  rename(Amino_Acid_Position_simple = Amino_Acid_Position_simple_CBE_YE1) %>%
  rename(phenotype = PHENOTYPES_CBE_YE1) %>%
  filter(Gene == "JAK1")

# BE3.9max_NGN
BE3.9max_NGN <- data_graphing_filtered %>% 
  mutate(editor = "BE3.9max-NGN (JAK1)") %>%
  mutate(zscore_FACS = NA) %>%
  select(Gene, guide, editor, zscore_iBE3_9NGN_cl12_ifng_vs_cntrl, zscore_FACS, Amino_Acid_Change_CBE, Amino_Acid_Position_CBE, Consequence_CBE, Clinical_Protein_Change_CBE, Citation_CBE, PTM_CBE, Amino_Acid_Position_simple_CBE, off_target_summary_NGG, off_target_summary_NGN, PHENOTYPES_CBE, sgRNA_ID) %>%
  rename(zscore_proliferation = zscore_iBE3_9NGN_cl12_ifng_vs_cntrl) %>%
  rename(Amino_Acid_Change = Amino_Acid_Change_CBE) %>%
  rename(Amino_Acid_Position = Amino_Acid_Position_CBE) %>%
  rename(Consequence = Consequence_CBE) %>%
  rename(Clinical_Protein_Change = Clinical_Protein_Change_CBE) %>%
  rename(Citation = Citation_CBE) %>%
  rename(PTM = PTM_CBE) %>%
  rename(Amino_Acid_Position_simple = Amino_Acid_Position_simple_CBE) %>%
  rename(phenotype = PHENOTYPES_CBE) %>%
  filter(Gene == "JAK1")

# ABE8e_NGN
ABE8e_NGN <- data_graphing_filtered %>% 
  mutate(editor = "ABE8e-NGN (JAK1)") %>%
  mutate(zscore_FACS = NA) %>%
  select(Gene, guide, editor, zscore_iABE8eNGN_cl12_ifng_vs_cntrl, zscore_FACS, Amino_Acid_Change_ABE, Amino_Acid_Position_ABE, Consequence_ABE, Clinical_Protein_Change_ABE, Citation_ABE, PTM_ABE, Amino_Acid_Position_simple_ABE, off_target_summary_NGG, off_target_summary_NGN, PHENOTYPES_ABE, sgRNA_ID) %>%
  rename(zscore_proliferation = zscore_iABE8eNGN_cl12_ifng_vs_cntrl) %>%
  rename(Amino_Acid_Change = Amino_Acid_Change_ABE) %>%
  rename(Amino_Acid_Position = Amino_Acid_Position_ABE) %>%
  rename(Consequence = Consequence_ABE) %>%
  rename(Clinical_Protein_Change = Clinical_Protein_Change_ABE) %>%
  rename(Citation = Citation_ABE) %>%
  rename(PTM = PTM_ABE) %>%
  rename(Amino_Acid_Position_simple = Amino_Acid_Position_simple_ABE) %>%
  rename(phenotype = PHENOTYPES_ABE) %>%
  filter(Gene == "JAK1")

BE3_NGG_IFNG_pathway <- as_tibble(read.csv("IFNG8_app_data.csv", stringsAsFactors = FALSE, header = TRUE))
BE3_NGG_IFNG_pathway <- mutate(BE3_NGG_IFNG_pathway, sgRNA_ID = as.character(sgRNA_ID))

app_data <- bind_rows(BE3_NGG, BE4max_YE1_NGN, BE3.9max_NGN, ABE8e_NGN)

app_data <- app_data %>% mutate(Amino_Acid_Position = trimws(Amino_Acid_Position, which = c("both"), whitespace = "[ \t\r\n]"))
app_data <- app_data %>% mutate(Amino_Acid_Position = str_squish(Amino_Acid_Position))
app_data <- app_data %>% mutate(Amino_Acid_Position = str_replace_all(Amino_Acid_Position, " ", ", ")) 

app_data <- bind_rows(app_data, BE3_NGG_IFNG_pathway)
app_data <- app_data %>% 
  select(!X) %>%
  mutate(cell_model = "HT-29")
CRC_9 <- as_tibble(read.csv("IFNG8_app_data_CRC9.csv", stringsAsFactors = FALSE, header = TRUE))
CRC_9 <- CRC_9 %>% 
  select(!X) %>%
  mutate(sgRNA_ID = as.character(sgRNA_ID))
app_data <- bind_rows(app_data, CRC_9)

app_data
app_data <- mutate(app_data, validation = ifelse(sgRNA_ID %in% c("908523459", "908523616", "908514537", "908514536", "908509432", "908510622", "908509431", "908510623", "908508937", "908509672", "908509010", "908512440", "908510034", "908510164", "908509007", "908510033", "908510274", "908509428", "908510028", "908510170", "908510032", "908508944", "908508948", "908510162"), "validated gRNA", "putative"))
write.csv(app_data, "input.csv")

```

```{r}
# Clinical outcome graph
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, is.na(Binarized_Clinical_Response_ABE) == FALSE), aes(x=Binarized_Clinical_Response_ABE, y= zscore_iABE8eNGN_cl12_ifng_vs_cntrl)) +
geom_boxplot(colour = "Salmon") +
geom_jitter(alpha=0.7, position = pos, size = 3) +
scale_colour_manual(values=c("orange", "royalblue1", "grey45","salmon", "black", "purple", "green3", "red")) +
labs(title= "Clinical Response to ICB: JAK1 NGN ABE screen", x = "", y = "zscore_iABE8eNGN_cl12_ifng_vs_cntrl") +
geom_text_repel(aes(label = paste0(Amino_Acid_Change_ABE, "\n", "(", Clinical_Protein_Change_ABE, Citation_ABE, ")")), size = 2, position = pos) +
theme_classic()
plot1

plot2 <- ggplot(subset(data_graphing_filtered, is.na(Binarized_Clinical_Response_CBE) == FALSE), aes(x=Binarized_Clinical_Response_CBE, y= zscore_iBE3_9NGN_cl12_ifng_vs_cntrl)) +
geom_boxplot(colour = "Salmon") +
geom_jitter(alpha=0.7, position = pos, size = 3) +
scale_colour_manual(values=c("orange", "royalblue1", "grey45","salmon", "black", "purple", "green3", "red")) +
labs(title= "Clinical Response to ICB: JAK1 NGN CBE screen", x = "", y = "zscore_iBE3_9NGN_cl12_ifng_vs_cntrl") +
geom_text_repel(aes(label = paste0(Amino_Acid_Change_CBE, "\n", "(", Clinical_Protein_Change_CBE, Citation_CBE, ")")), size = 2, position = pos) +
theme_classic()
plot2

plot3 <- ggplot(subset(data_graphing_filtered, is.na(Binarized_Clinical_Response_CBE_YE1) == FALSE), aes(x=Binarized_Clinical_Response_CBE_YE1, y= zscore_IFNG_Cntrl_prolif_average_NG_BE4)) +
geom_boxplot(colour = "Salmon") +
geom_jitter(alpha=0.7, position = pos, size = 3) +
scale_colour_manual(values=c("orange", "royalblue1", "grey45","salmon", "black", "purple", "green3", "red")) +
labs(title= "Clinical Response to ICB: JAK1 NGN CBE YE1 screen", x = "", y = "zscore_IFNG_Cntrl_prolif_average_NG_BE4") +
geom_text_repel(aes(label = paste0(Amino_Acid_Change_CBE_YE1, "\n", "(", Clinical_Protein_Change_CBE_YE1, Citation_CBE_YE1, ")")), size = 2, position = pos) +
theme_classic()
plot3

plot4 <- ggplot(subset(data_graphing_filtered, is.na(Binarized_Clinical_Response_CBE) == FALSE), aes(x=Binarized_Clinical_Response_CBE, y= zscore_IFNG_Proliferation_Average_NGG)) +
geom_boxplot(colour = "Salmon") +
geom_jitter(alpha=0.7, position = pos, size = 3) +
scale_colour_manual(values=c("orange", "royalblue1", "grey45","salmon", "black", "purple", "green3", "red")) +
labs(title= "Clinical Response to ICB: JAK1 NGG CBE screen", x = "", y = "zscore_IFNG_Proliferation_Average_NGG") +
geom_text_repel(aes(label = paste0(Amino_Acid_Change_CBE, "\n", "(", Clinical_Protein_Change_CBE, Citation_CBE, ")")), size = 2, position = pos) +
theme_classic()
plot4
```

```{r}
# boxplot stats
data_graphing_filtered %>% filter(is.na(zscore_iABE8eNGN_cl12_ifng_vs_cntrl)==FALSE) %>% count(Consequence_ABE)
data_graphing_filtered %>% filter(is.na(zscore_iBE3_9NGN_cl12_ifng_vs_cntrl)==FALSE) %>% count(Consequence_CBE)
data_graphing_filtered %>% filter(is.na(zscore_IFNG_Cntrl_prolif_average_NG_BE4)==FALSE) %>% count(Consequence_CBE_YE1)
data_graphing_filtered %>% filter(is.na(zscore_IFNG_Proliferation_Average_NGG)==FALSE) %>% count(Consequence_CBE)

mean_NT_A <- data_graphing_filtered %>% 
filter(Consequence_ABE == "non-targeting") %>%
select(zscore_iABE8eNGN_cl12_ifng_vs_cntrl)
mean_splice_A <- data_graphing_filtered %>% 
filter(Consequence_ABE == "splice variant") %>%
select(zscore_iABE8eNGN_cl12_ifng_vs_cntrl)

mean_NT_C <- data_graphing_filtered %>% 
filter(Consequence_CBE == "non-targeting") %>%
select(zscore_iBE3_9NGN_cl12_ifng_vs_cntrl)
mean_splice_C <- data_graphing_filtered %>% 
filter(Consequence_CBE == "splice variant") %>%
select(zscore_iBE3_9NGN_cl12_ifng_vs_cntrl)

mean_NT_CY <- data_graphing_filtered %>% 
filter(Consequence_CBE_YE1 == "non-targeting") %>%
select(zscore_IFNG_Cntrl_prolif_average_NG_BE4)
mean_splice_CY <- data_graphing_filtered %>% 
filter(Consequence_CBE_YE1 == "splice variant") %>%
select(zscore_IFNG_Cntrl_prolif_average_NG_BE4)

mean_NT_CNGG <- data_graphing_filtered %>% 
filter(Consequence_CBE == "non-targeting") %>%
select(zscore_IFNG_Proliferation_Average_NGG)
mean_splice_CNGG <- data_graphing_filtered %>% 
filter(Consequence_CBE == "splice variant") %>%
select(zscore_IFNG_Proliferation_Average_NGG)

mean_missense_A <- data_graphing_filtered %>% 
filter(Consequence_ABE == "missense") %>%
select(zscore_iABE8eNGN_cl12_ifng_vs_cntrl)
mean_missense_proline_A <- data_graphing_filtered %>% 
filter(Consequence_ABE == "missense proline") %>%
select(zscore_iABE8eNGN_cl12_ifng_vs_cntrl)

t.test(mean_NT_A, mean_splice_A, var.equal = TRUE)
t.test(mean_NT_C, mean_splice_C, var.equal = TRUE)
t.test(mean_NT_CY, mean_splice_CY, var.equal = TRUE)
t.test(mean_NT_CNGG, mean_splice_CNGG, var.equal = TRUE)

t.test(mean_missense_A, mean_missense_proline_A, var.equal = TRUE)

```

```{r}
# boxplots
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# factor levels
data_graphing_filtered$Consequence_CBE <- factor(data_graphing_filtered$Consequence_CBE, levels = (c("UTR", "promoter", "synonymous", "intergenic", "non-targeting", "start lost", "NA", "splice variant", "stop codon", "missense proline", "missense", "stop essential", "stop non-essential")))
data_graphing_filtered$Consequence_CBE_YE1 <- factor(data_graphing_filtered$Consequence_CBE_YE1, levels = (c("UTR", "promoter", "synonymous", "intergenic", "non-targeting", "start lost", "NA", "splice variant", "stop codon", "missense proline", "missense", "stop essential", "stop non-essential")))
data_graphing_filtered$Consequence_ABE <- factor(data_graphing_filtered$Consequence_ABE, levels = (c("UTR", "promoter", "synonymous", "intergenic", "non-targeting", "start lost", "NA", "splice variant", "stop codon", "missense proline", "missense")))

plot1 <- ggplot(subset(data_graphing_filtered, Consequence_ABE %in% c("missense", "missense proline", "splice variant", "stop codon", "UTR", "promoter", "synonymous", "intergenic", "non-targeting")), aes(y=Consequence_ABE, x= zscore_iABE8eNGN_cl12_ifng_vs_cntrl, colour = Consequence_ABE)) + 
geom_jitter(position = pos, shape = 16, alpha = 0.30, size =3) +
geom_boxplot(colour = "salmon", alpha = 0) +
scale_colour_manual(values=c("grey65", "grey65", "grey65", "grey65", "grey65", "orange", "springgreen3", "black", "grey65", "grey65"), guide="none") +
labs(title= "ABE8e-NGN", x = "proliferation z-score", y = "") +
annotate("text", y = 1:8, x = -2.5, label = c("italic(n) == 955", "italic(n) == 108", "italic(n) == 140", "italic(n) == 158", "italic(n) == 55", "italic(n) == 227", "italic(n) == 292", "italic(n) == 550"), parse = TRUE) +
annotate("text", y = 6, x = 8, label = c("P = 1.41x10-10"), parse = FALSE) +
theme_classic()
plot1

plot2 <- ggplot(subset(data_graphing_filtered, Consequence_CBE %in% c("missense", "missense proline", "splice variant", "stop codon", "UTR", "promoter", "synonymous", "intergenic", "non-targeting")), aes(y=Consequence_CBE, x= zscore_iBE3_9NGN_cl12_ifng_vs_cntrl, colour = Consequence_CBE)) + 
geom_jitter(position = pos, shape = 16, alpha = 0.30, size =3) +
geom_boxplot(colour = "salmon", alpha = 0) +
scale_colour_manual(values=c("grey65", "grey65", "grey65", "grey65", "grey65", "orange", "red2", "black", "grey65", "grey65"), guide="none") +
labs(title= "BE3.9max-NGN", x = "proliferation z-score", y = "") +
annotate("text", y = 1:8, x = -2.5, label = c("italic(n) == 566", "italic(n) == 119", "italic(n) == 341", "italic(n) == 158", "italic(n) == 55", "italic(n) == 225", "italic(n) == 81", "italic(n) == 859"), parse = TRUE) +
annotate("text", y = 6, x = 8, label = c("P = 2.42x10-6"), parse = FALSE) +
theme_classic()
plot2

plot3 <- ggplot(subset(data_graphing_filtered, Consequence_CBE_YE1 %in% c("missense", "missense proline", "splice variant", "stop codon", "UTR", "promoter", "synonymous", "intergenic", "non-targeting")), aes(y=Consequence_CBE_YE1, x= zscore_IFNG_Cntrl_prolif_average_NG_BE4, colour = Consequence_CBE_YE1)) + 
geom_jitter(position = pos, shape = 16, alpha = 0.30, size =3) +
geom_boxplot(colour = "salmon", alpha = 0) +
scale_colour_manual(values=c("grey65", "grey65", "grey65", "grey65", "grey65", "orange", "red2", "black", "grey65", "grey65"), guide="none") +
labs(title= "BE4max-YE1-NGN", x = "proliferation z-score", y = "") +
annotate("text", y = 1:8, x = -3.5, label = c("italic(n) == 398", "italic(n) == 83", "italic(n) == 311", "italic(n) == 158", "italic(n) == 55", "italic(n) == 85", "italic(n) == 47", "italic(n) == 564"), parse = TRUE) +
annotate("text", y = 6, x = 16, label = c("P = 2.01x10-6"), parse = FALSE) +
theme_classic()
plot3

plot4 <- ggplot(subset(data_graphing_filtered, Consequence_CBE %in% c("missense", "missense proline", "splice variant", "stop codon", "UTR", "promoter", "synonymous", "intergenic", "non-targeting")), aes(y=Consequence_CBE, x= zscore_IFNG_Proliferation_Average_NGG, colour = Consequence_CBE)) + 
geom_jitter(position = pos, shape = 16, alpha = 0.30, size =3) +
geom_boxplot(colour = "salmon", alpha = 0) +
scale_colour_manual(values=c("grey65", "grey65", "grey65", "grey65", "grey65", "orange", "red2", "black", "grey65", "grey65"), guide="none") +
labs(title= "BE3-NGG", x = "proliferation z-score", y = "") +
annotate("text", y = 1:8, x = -3.5, label = c("italic(n) == 134", "italic(n) == 114", "italic(n) == 79", "italic(n) == 158", "italic(n) == 55", "italic(n) == 71", "italic(n) == 17", "italic(n) == 216"), parse = TRUE) +
annotate("text", y = 6, x = 16, label = c("P = 9.37x10-5"), parse = FALSE) +
theme_classic()
plot4
```

```{r}
# list of top hits in the NGN screens for supp table 5

supp_table_5 <- app_data %>%
  filter(zscore_proliferation > 3 & Consequence == "missense" | zscore_proliferation > 3 & Consequence == "missense proline" | zscore_proliferation < -3 & Consequence == "missense" | zscore_proliferation < -3 & Consequence == "missense proline" | zscore_FACS > 3 & Consequence == "missense") %>%
  distinct() %>%
  mutate(predicted_function = ifelse(zscore_proliferation > 3 & Gene != "SOCS1", "LOF",
                                     ifelse(zscore_FACS > 3 & Gene != "SOCS1", "LOF", 
                                            ifelse(zscore_proliferation < -3 & Gene == "SOCS1", "LOF",
                                                   "GOF")))) %>%
  mutate(predicted_function = ifelse(is.na(predicted_function) == TRUE, "GOF", predicted_function))
write.csv(supp_table_5, "Supplementary_Table_5.csv")

supp_table_5 %>% select(Gene, Amino_Acid_Position, cell_model) %>%
  distinct() %>%
  count(Gene)

supp_table_5 %>% select(Gene, Amino_Acid_Position, cell_model) %>%
  distinct()

supp_table_5 %>% select(Gene, Amino_Acid_Position, predicted_function, cell_model) %>%
  distinct() %>%
  count(Gene, predicted_function, cell_model)

supp_table_5 %>% select(Gene, Amino_Acid_Position, cell_model) %>%
  distinct() %>%
  count()

supp_table_5 %>% select(Gene, Amino_Acid_Position, predicted_function, cell_model) %>%
  distinct() %>%
  count(predicted_function)

supp_table_5 %>% select(Gene, Amino_Acid_Position, editor) %>%
  distinct() %>%
  count(editor)

supp_table_5 %>% select(Gene, Amino_Acid_Position, predicted_function, cell_model) %>%
  distinct() %>%
  count(predicted_function)

supp_table_5 %>% 
  filter(cell_model =="HT-29") %>%
  mutate(Variant_Recapitulation = ifelse(str_detect(Clinical_Protein_Change, Amino_Acid_Change) & Citation != "gnomAD_v3.1", TRUE, 
                                         ifelse(str_detect(Amino_Acid_Change, Clinical_Protein_Change) & Citation != "gnomAD_v3.1", TRUE, FALSE))) %>%
  select(Gene, Amino_Acid_Change, Clinical_Protein_Change, Variant_Recapitulation, editor, cell_model, Citation) %>%
  distinct() 

supp_table_5 %>% 
  filter(cell_model =="HT-29") %>%
  mutate(Variant_Recapitulation = ifelse(str_detect(Clinical_Protein_Change, Amino_Acid_Change) & Citation != "gnomAD_v3.1", TRUE, 
                                         ifelse(str_detect(Amino_Acid_Change, Clinical_Protein_Change) & Citation != "gnomAD_v3.1", TRUE, FALSE))) %>%
  select(Gene, Amino_Acid_Change, Clinical_Protein_Change, Variant_Recapitulation, editor, cell_model, Citation) %>%
  distinct() %>%
    count(editor, Variant_Recapitulation, cell_model)

supp_table_5 %>%
  filter(cell_model == "HT-29") %>%
  select(Amino_Acid_Position, Amino_Acid_Change, Clinical_Protein_Change, Citation, editor) %>%
    distinct

supp_table_5 %>%
  filter(cell_model == "HT-29", editor == "BE3.9max-NGN (JAK1)") %>%
  select(Amino_Acid_Position, Amino_Acid_Change, Clinical_Protein_Change, Citation) %>%
    distinct

supp_table_5 %>%
  filter(cell_model == "HT-29", editor == "BE3.9max-NGN (JAK1)") %>%
  select(Amino_Acid_Position, Amino_Acid_Change, Clinical_Protein_Change, Citation) %>%
    distinct %>%
  count(Citation) %>% write.csv("BENGN.csv")

supp_table_5%>%
  filter(cell_model == "HT-29", editor == "ABE8e-NGN (JAK1)") %>%
  select(Amino_Acid_Position, Amino_Acid_Change, Clinical_Protein_Change, Citation) %>%
    distinct 

supp_table_5%>%
  filter(cell_model == "HT-29", editor == "ABE8e-NGN (JAK1)") %>%
  select(Amino_Acid_Position, Amino_Acid_Change, Clinical_Protein_Change, Citation) %>%
    distinct %>%
  count(Citation) %>% write.csv("ABENGN.csv")

#CBE NGN clinical match
14/(14+26+4) *100


#ABE NGN clinical match
10/(10+114+53) *100

```
```{r}
#overalp between CRC-9 and HT-29
supp_table_5 
supp_table_5 %>% select(cell_model, editor, sgRNA_ID) %>%
  filter(editor == "BE3-NGG (IFNG pathway)") %>%
  distinct()

supp_table_5 %>% select(editor, sgRNA_ID) %>%
  filter(editor == "BE3-NGG (IFNG pathway)") %>%
  distinct()

HT29_hits <- supp_table_5 %>% 
  select(editor, sgRNA_ID, cell_model) %>%
  filter(editor == "BE3-NGG (IFNG pathway)" & cell_model == "HT-29") %>%
  select(sgRNA_ID) %>%
  distinct()

CRC9_hits <- supp_table_5 %>% 
  select(editor, sgRNA_ID, cell_model) %>%
  filter(editor == "BE3-NGG (IFNG pathway)" & cell_model == "CRC-9_organoid") %>%
  select(sgRNA_ID) %>%
  distinct()

HT29_hits
CRC9_hits
combined_hits <- bind_rows(HT29_hits, CRC9_hits) %>% distinct()
combined_hits
anti_join(HT29_hits, CRC9_hits)
anti_join(CRC9_hits, HT29_hits)
# 61
# 71
```

```{r}
# dot plots _ proliferation - JAK1
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(app_data, Consequence %in% c("intron", "UTR", "promoter", NA) ==FALSE & Gene =="JAK1"), aes(x=Amino_Acid_Position_simple, y= zscore_proliferation, colour = editor, shape = Consequence)) + 
geom_point(position = pos, size = 2, alpha = 0.65) +
labs(title= "JAK1 base editing screens", x="amino acid position", y = "z-score: proliferation screens", shape = "consequence", colour = "base editor screen") +
scale_colour_manual(values=c("black", "springgreen4", "orange", "royal blue", "red")) +
scale_shape_manual(values=c(19, 1, 6, 4, 5, 3)) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(zscore_proliferation > 3 & str_detect(Amino_Acid_Position, "108|887|590|775|655|1034|182|1021|1023|1024|1003") | zscore_proliferation < -2 & str_detect(Amino_Acid_Position, "724|658|1099|1103|267"), paste(Amino_Acid_Position), "")), size = 3, position = pos) +
annotate("rect", xmin =34, xmax = 420, ymin = -2.5, ymax = 2, alpha = 0.1) +
annotate("text", x =220, y = -3, label = "FERM", size = 4.5, colour = "grey45") +
annotate("rect", xmin =439, xmax = 544, ymin = -2.5, ymax = 2, alpha = 0.1) +
annotate("text", x =500, y = -3, label = "SH2-like", size = 4.5, colour = "grey45") +
annotate("rect", xmin =583, xmax = 855, ymin = -2.5, ymax = 2, alpha = 0.1) +
annotate("text", x =710, y = -3, label = "pseudokinase", size = 4.5, colour = "grey45") +
annotate("rect", xmin =875, xmax = 1153, ymin = -2.5, ymax = 2, alpha = 0.1) +
annotate("text", x =1000, y = -3, label = "kinase", size = 4.5, colour = "grey45") +
theme_classic()
plot1
```

```{r}
#  geom_rug
pos = position_jitter(width = 0.5, height = 0.5, seed = 2)

plot1 <- ggplot(subset(app_data, Consequence %in% c("missense") & Gene =="JAK1" & (zscore_proliferation >3 | zscore_proliferation < -3)), aes(x=Amino_Acid_Position_simple, colour = zscore_proliferation)) +
theme_classic() +
labs(title= "JAK1 missense hits") +
geom_rug(length = unit(1, "npc")) +
ylim(0, 0.05) +
theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
scale_color_viridis_c()
plot1
```

```{r}
# read in cancer cell lines with JAK1 mutations
cell_lines <- as_tibble(read.csv("mutations_all_20211208.csv", stringsAsFactors = FALSE, header = TRUE))

cell_lines <-rename(cell_lines, Amino_Acid_Change = protein_mutation)

cell_lines <- mutate(cell_lines, Amino_Acid_Change = str_remove(Amino_Acid_Change, "p."))
cell_lines

# make COSMIC AA.Mutations into three letter code for compatibility
cell_lines <- mutate(cell_lines, Amino_Acid_Change = str_replace_all(string = Amino_Acid_Change, 
                c("A" = "Ala",
                  "R" = "Arg",
                  "N" = "Asn",
                  "D" = "Asp",
                  "C" = "Cys",
                  "G" = "Gly",
                  "E" = "Glu",
                  "Q" = "Gln",
                  "H" = "His",
                  "I" = "Ile",
                  "L" = "Leu",
                  "K" = "Lys",
                  "M" = "Met",
                  "P" = "Pro",
                  "F" = "Phe",
                  "S" = "Ser",
                  "T" = "Thr",
                  "W" = "Trp",
                  "Y" = "Tyr",
                  "V" = "Val",
                  "\\*" = "Ter")))

cell_lines <-mutate(cell_lines, Amino_Acid_Position= as.numeric(str_extract(Amino_Acid_Change, "[[:digit:]]+")))
cell_lines<-rename(cell_lines, Amino_Acid_Change_cell_line = Amino_Acid_Change)
cell_lines <- mutate(cell_lines, Amino_Acid_Position = as.character(Amino_Acid_Position))
cell_lines<-rename(cell_lines, Amino_Acid_Position_cell_line = Amino_Acid_Position)
cell_lines<-rename(cell_lines, Gene = gene_symbol)
cell_lines<-rename(cell_lines, Consequence = effect)
cell_lines<-rename(cell_lines, Institute = source)
cell_lines<-rename(cell_lines, VAF = vaf)
cell_lines<-rename(cell_lines, cell_line = model_name)

cell_lines 
supp_table_5
supp_table_5_cell_line <- left_join(supp_table_5, cell_lines)
supp_table_5_cell_line

#homozygous LOF missense
supp_table_5_cell_line %>% 
  filter(Gene == "JAK1") %>%
  filter(is.na(cell_line) == FALSE) %>%
  filter(str_detect(Amino_Acid_Position, Amino_Acid_Position_cell_line)) %>%
  select(cell_line, Amino_Acid_Change, Amino_Acid_Change_cell_line, Amino_Acid_Position_cell_line, VAF, Institute) %>%
  filter(VAF > 0.75) %>%
  distinct()

# no of cell lines
cell_lines %>% select(model_id) %>% distinct()
cell_lines %>% select(cell_line) %>% distinct()
1357

# complete LOF  homozygous nonsense or frameshift
cell_lines %>%
  filter(Gene == "JAK1") %>%
  filter(Consequence %in% c("frameshift", "nonsense")) %>%
  filter(VAF > 0.75) %>%
  select(cell_line) %>%
  distinct()
10/1357*100

cell_lines %>% 
  filter(Gene == "JAK1") %>%
  filter(Consequence %in% c("missense")) %>%
  filter(VAF > 0.75) %>%
  distinct()

3/1357*100

#GOF
supp_table_5_cell_line %>% 
  filter(Gene == "JAK1") %>%
  filter(is.na(cell_line) == FALSE) %>%
  filter(str_detect(Amino_Acid_Position, Amino_Acid_Position_cell_line)) %>%
  filter(predicted_function == "GOF") %>%
  select(cell_line, Amino_Acid_Change, Amino_Acid_Change_cell_line, Amino_Acid_Position_cell_line, VAF, Institute) %>%
  distinct()

```
