---
title: "JAK1"
author: "Matt Coelho"
date: "07/06/2020"
output: html_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# load packages
library(tidyverse)
library(reshape2)
library(ggrepel)
library(stringr)
```

```{r}
# read in guides for JAK1 from WGE
data <- as_tibble(read.csv("bedit_jak1_sgrnas_v1.1_annotated_library.csv", stringsAsFactors = FALSE, header = TRUE))
```

```{r}
# determine window for base editor (here is 4-9). This is lenient and includes guides with G mismatch at pos 1.
data <-  mutate(data, window = ifelse(strand == "+", substr(seq, 4, 9), substr(seq, 15, 20)))

# edit window (here C to T for CBE)
data <-  mutate(data, edit = ifelse(strand == "+", str_replace_all(window, "C", "t"), str_replace_all(window, "G", "a")))

# make mutant seq (optional making full mutant protospacer genotype)
data <-  mutate(data, mutant = ifelse(strand == "+", paste0(substr(seq, 1, 3), edit, substr(seq, 10, 23)), paste0(substr(seq, 1, 14), edit, substr(seq, 21, 23))))
```

```{r}
# give numbers of each gRNA_type in the library
data %>% count(sgRNA_type)
```

```{r}
# add a FORWARD guide column (reverse reverse complement for - strand seq)
# add 5' G for non-G starting guides. You can use this to ORDER GUIDE libraries (add appropriate Gibson cloning adapters)
# get rid of PAM
data <- mutate(data, guide = ifelse(strand == "+", substr(seq, 1, 20), substr(seq, 4, 23)))

# complement
data <- mutate(data, guide = ifelse(strand == "+", guide, str_replace_all(guide, "A", "t")))
data <- mutate(data, guide = ifelse(strand == "+", guide, str_replace_all(guide, "T", "a")))
data <- mutate(data, guide = ifelse(strand == "+", guide, str_replace_all(guide, "C", "g")))
data <- mutate(data, guide = ifelse(strand == "+", guide, str_replace_all(guide, "G", "c")))
data <- mutate(data, guide = str_to_upper(guide))

# reverse
data <- mutate(data, guide = ifelse(strand == "+", guide, paste0(substr(guide, 20, 20), substr(guide, 19, 19),substr(guide, 18, 18),substr(guide, 17, 17),substr(guide, 16, 16),substr(guide, 15, 15), substr(guide, 14, 14), substr(guide, 13, 13), substr(guide, 12, 12), substr(guide, 11, 11), substr(guide, 10, 10), substr(guide, 9, 9), substr(guide, 8, 8), substr(guide, 7, 7), substr(guide, 6, 6), substr(guide, 5, 5), substr(guide, 4, 4), substr(guide, 3, 3), substr(guide, 2, 2), substr(guide, 1, 1))))

# add a G for the expression from U6 promtoter by replaceing the base at pos. 1 if, it is not a G
data <- mutate(data, G_guide = ifelse(substr(guide, 1, 1) == "G", guide, paste0("G", substr(guide, 2, 20))))

# primers
# forward_primer
data <- mutate(data, for_primer = paste0("CACC", G_guide))

# reverse_primer
data <- mutate(data, rev_primer = G_guide)
data <- mutate(data, rev_primer = str_replace_all(rev_primer, "A", "t"))
data <- mutate(data, rev_primer = str_replace_all(rev_primer, "T", "a"))
data <- mutate(data, rev_primer = str_replace_all(rev_primer, "C", "g"))
data <- mutate(data, rev_primer = str_replace_all(rev_primer, "G", "c"))
data <- mutate(data, rev_primer = str_to_upper(rev_primer))

data <- mutate(data, rev_primer = paste0(substr(rev_primer, 20, 20), substr(rev_primer, 19, 19),substr(rev_primer, 18, 18),substr(rev_primer, 17, 17),substr(rev_primer, 16, 16),substr(rev_primer, 15, 15), substr(rev_primer, 14, 14), substr(rev_primer, 13, 13), substr(rev_primer, 12, 12), substr(rev_primer, 11, 11), substr(rev_primer, 10, 10), substr(rev_primer, 9, 9), substr(rev_primer, 8, 8), substr(rev_primer, 7, 7), substr(rev_primer, 6, 6), substr(rev_primer, 5, 5), substr(rev_primer, 4, 4), substr(rev_primer, 3, 3), substr(rev_primer, 2, 2), substr(rev_primer, 1, 1)))

data <- mutate(data, rev_primer = paste0("AAAC", rev_primer))
```

```{r}
# BE4 vs BE3 windows- BE3_window_only, considering G mismatch as not counting in the window
data <- mutate(data, BE3_window_positions = 
                 ifelse(substr(guide, 1, 1) == "G" & substr(guide, 4, 4) == "C" | substr(guide, 1, 1) == "G" & substr(guide, 8, 8) == "C" | substr(guide, 1, 1) == "G" & substr(guide, 9, 9) == "C", TRUE, 
                                                   ifelse(substr(guide, 1, 1) != "G" & substr(guide, 4, 4) == "C" | substr(guide, 1, 1) != "G" & substr(guide, 5, 5) == "C" | substr(guide, 1, 1) != "G" & substr(guide, 9, 9) == "C", TRUE, FALSE)))

data %>% select(strand, guide, mutant, BE3_window_positions)
```

```{r}
# amplicon seq analysis

amplicon_seq <- as_tibble(read.csv("ampliconseq_conditions_230721.csv", stringsAsFactors = FALSE, header = TRUE))

data_amplicon <- data %>%
  select(sgRNA_ID, Gene, chr, start, end, strand, guide)

amplicon_seq <- mutate(amplicon_seq, sgRNA_ID = as.character(sgRNA_ID))

amplicon_seq2 <- left_join(amplicon_seq, data_amplicon)

write.csv(amplicon_seq2, "ampliconseq_conditions_3_260721.csv")
```

```{r}
# add on target score Azimuth 2.0 (Rule Set 2 - CRISPick, Doench 2016)
Rule_Set_2 <- as_tibble(read.csv("JAK1_exonic_RuleSet2_CRISPick.csv", stringsAsFactors = FALSE, header = TRUE))
data <- left_join(data, Rule_Set_2)
data <- mutate(data, Rule_Set_2_High = ifelse(Rule_Set_2_Score > 0.5, TRUE, FALSE))
data %>% filter(Gene == "JAK1" & sgRNA_type == "exonic") %>%
  select(Rule_Set_2_Score, Gene, sgRNA_type, guide, Rule_Set_2_High)
data %>% filter(sgRNA_type == "stop_essential") %>%
  select(Rule_Set_2_Score, Gene, sgRNA_type, guide, Rule_Set_2_High)
```

```{r}
# read in raw L2FC values
L2FC_values <- as_tibble(read.csv("read_counts_2473_121120.csv", stringsAsFactors = FALSE, header = TRUE))
data <- left_join(data, L2FC_values)

# compute mean L2FC
data <- mutate(data, Average_L2FC_Proliferation = ((L2FC_Proliferation_1 + L2FC_Proliferation_2)/2))
data <- mutate(data, Average_L2FC_FACS = ((L2FC_FACS_1 + L2FC_FACS_2)/2))
```

```{r}
# read in zscores
zscores <- as_tibble(read.csv("160321_JAK1_zscores.csv", stringsAsFactors = FALSE, header = TRUE))
data <- left_join(data, zscores)
```

```{r}
# Rule_Set_2 graphs 
plot1 <- ggplot(subset(data, is.na(Rule_Set_2_Score) == FALSE & sgRNA_type == "exonic"), aes(x=Rule_Set_2_Score, y= zscore_IFNG_Proliferation_Average)) +
geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): Rule Set 2 Score: JAK1 Exonic gRNAs") +
geom_smooth(method = lm) +
theme_classic()
plot1

plot2 <- ggplot(subset(data, is.na(Rule_Set_2_Score) == FALSE & sgRNA_type == "exonic"), aes(x=Rule_Set_2_High, y= zscore_IFNG_Proliferation_Average)) +
geom_boxplot(colour = "salmon") +
geom_jitter(alpha=0.1) +
labs(title= "Rule Set 2 Score: JAK1 Exonic gRNAs", x="Rule Set 2 Score", y = "z-score: control vs IFNG") +
scale_x_discrete(labels=c("FALSE" = "<0.5", "TRUE" = ">0.5")) +
theme_classic() +
annotate("text", x = 1.5, y = 11, label = c("P= 0.001752")) +
annotate("segment", x = 1.1, xend = 1.9, y = 10, yend = 10, alpha = 0.7) +
theme(axis.text.x = element_text(size = rel(1.1)))
plot2

plot3 <- ggplot(subset(data, is.na(Rule_Set_2_Score) == FALSE & sgRNA_type == "stop_essential"), aes(x=Rule_Set_2_Score, y= zscore_T0_Average)) +
geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): Rule Set 2 Score: Stop Essential vs T0") +
geom_smooth(method = lm) +
theme_classic()
plot3

plot4 <- ggplot(subset(data, is.na(Rule_Set_2_Score) == FALSE & sgRNA_type == "stop_essential"), aes(x=Rule_Set_2_High, y= zscore_T0_Average)) +
geom_boxplot(colour = "salmon") +
geom_jitter(alpha=0.1) +
labs(title= "Rule Set 2 Score: stop-essential gRNAs", x="Rule Set 2 Score", y = "z-score: control vs T0") +
scale_x_discrete(labels=c("FALSE" = "<0.5", "TRUE" = ">0.5")) +
theme_classic() +
annotate("text", x = 1.5, y = 6, label = c("P= 0.0009023")) +
annotate("segment", x = 1.1, xend = 1.9, y = 5, yend = 5, alpha = 0.7) +
theme(axis.text.x = element_text(size = rel(1.1)))
plot4

###
mean_exonic_Score_High_TRUE <- data %>% 
filter(sgRNA_type == "exonic", Rule_Set_2_High == TRUE) %>%
select(zscore_IFNG_Proliferation_Average) %>%
as.data.frame(mean_exonic_Score_High_TRUE)

mean_exonic_Score_High_FALSE <- data %>% 
filter(sgRNA_type == "exonic", Rule_Set_2_High == FALSE) %>%
select(zscore_IFNG_Proliferation_Average) %>%
as.data.frame(mean_exonic_Score_High_FALSE)

t.test(mean_exonic_Score_High_TRUE, mean_exonic_Score_High_FALSE, var.equal = TRUE)

###
mean_stop_essential_Score_High_TRUE <- data %>% 
filter(sgRNA_type == "stop_essential", Rule_Set_2_High == TRUE) %>%
select(zscore_T0_Average) %>%
as.data.frame(mean_stop_essential_Score_High_TRUE)

mean_stop_essential_Score_High_FALSE <- data %>% 
filter(sgRNA_type == "stop_essential", Rule_Set_2_High == FALSE) %>%
select(zscore_T0_Average) %>%
as.data.frame(mean_stop_essential_Score_High_FALSE)

t.test(mean_stop_essential_Score_High_TRUE, mean_stop_essential_Score_High_FALSE, var.equal = TRUE)
```

```{r}
# quality control plots
plot1 <- ggplot(subset(data, sgRNA_type == "promoter"| sgRNA_type == "exonic"| sgRNA_type == "intergenic"), aes(x=start, y= zscore_IFNG_FACS_1, colour = sgRNA_type)) + geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1 promoter")
plot1

plot2 <- ggplot(subset(data, sgRNA_type == "promoter"| sgRNA_type == "exonic"| sgRNA_type == "intergenic"), aes(x=start, y= zscore_IFNG_FACS_2, colour = sgRNA_type)) + geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1 promoter")
plot2

plot3 <- ggplot(subset(data, sgRNA_type == "promoter"| sgRNA_type == "exonic"| sgRNA_type == "intergenic"), aes(x=start, y= zscore_IFNG_Proliferation_1, colour = sgRNA_type)) + geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1 promoter")
plot3

plot4 <- ggplot(subset(data, sgRNA_type == "promoter"| sgRNA_type == "exonic"| sgRNA_type == "intergenic"), aes(x=start, y= zscore_IFNG_Proliferation_2, colour = sgRNA_type)) + geom_jitter(size=3, alpha=0.8, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1 promoter")
plot4

plot5 <- ggplot(data, aes(y= zscore_T0_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(y= zscore_T0_2, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_classic()
plot6

plot7 <- ggplot(data, aes(y= zscore_T0_Average, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_classic()
plot7

plot8 <- ggplot(data, aes(y= T0_Reads_1)) + 
geom_histogram() +
labs(title= "Reads_1")
plot8

plot9 <- ggplot(data, aes(y= T0_Reads_2)) + 
geom_histogram() +
labs(title= "Reads_2")
plot9

plot10 <- ggplot(data, aes(x= zscore_T0_Average, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "z-score: control vs T0 ", colour = "gRNA type", fill = "gRNA type") +
theme_classic()
plot10
```

```{r}
# all_hits_L2FC
data %>% filter(L2FC_Proliferation_1 >1 | L2FC_Proliferation_2 >1 | L2FC_Proliferation_1 < -0.5 | L2FC_Proliferation_2 < -0.5 | L2FC_FACS_1 >1 | L2FC_FACS_2 >1) %>% 
  filter(Gene == "JAK1") %>%
  select(Gene, sgRNA_ID, G_guide, sgRNA_type, off_target_summary, zscore_IFNG_Proliferation_Average, zscore_IFNG_FACS_Average, Average_L2FC_Proliferation, Average_L2FC_FACS) %>%
  distinct()
```

```{r}
# all_hits_zscores
all_hits_zscores <- data %>% filter(zscore_IFNG_Proliferation_Average > 5  | zscore_IFNG_Proliferation_Average < -2.5 | zscore_IFNG_FACS_Average > 5) %>% 
  filter(Gene == "JAK1") %>%
  select(Gene, sgRNA_ID, G_guide, sgRNA_type, off_target_summary, zscore_IFNG_Proliferation_Average, zscore_IFNG_FACS_Average, Average_L2FC_Proliferation, Average_L2FC_FACS) %>%
  distinct()
```

```{r}
# promoter gRNAs dot plots
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data, sgRNA_type == "promoter"), aes(x=start, y= zscore_IFNG_Proliferation_Average)) +
geom_point(size=3, position = pos, alpha = 0.7) +
labs(title= "BE edit prediction (C>T): JAK1 promoter", x = "genomic pos: chr 1 (GRCh38)", y = "z-score: control vs IFNG\nProliferation screen") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
theme_bw()
plot1
```

```{r}
# replicate correlation
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data, Gene == "JAK1"), aes(x=zscore_IFNG_Proliferation_1, y= zscore_IFNG_Proliferation_2, colour = zscore_IFNG_FACS_Average)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all JAK1 gRNAs", x = "z-score: proliferation screen exp. 1", y = "z-score: proliferation screen exp. 2", colour = "z-score: FACS screens") +
geom_smooth(method = lm) +
geom_text_repel(aes(label=ifelse(sgRNA_ID == "908523459", "5' UTR variant", ifelse(sgRNA_ID == "908523616", "promoter variant", ""))), size = 3, position = pos) +
theme_classic()
plot1

plot2 <- ggplot(subset(data, Gene == "JAK1"), aes(x=zscore_IFNG_FACS_1, y= zscore_IFNG_FACS_2, colour = zscore_IFNG_Proliferation_Average)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all JAK1 gRNAs", x = "z-score: FACS screen exp. 1", y = "z-score: FACS screen exp. 2", colour = "z-score: proliferation screens") +
geom_smooth(method = lm) +
theme_classic()
plot2
```

```{r}
# find positions of where edit is preceded by a G
data <- mutate(data, G_preceding_edit = ifelse(str_detect(mutant, "Gt") | str_detect(mutant, "aC") == TRUE, TRUE, FALSE))
#filter data for only guides that edit
data_G <- filter(data, window!=edit)
```

```{r}
# G_preceding plot
plot1 <- ggplot(subset(data_G, sgRNA_type == "exonic"), aes(y=zscore_IFNG_Proliferation_Average, x= G_preceding_edit)) + 
geom_boxplot(colour = "Salmon") +
geom_jitter(alpha = 0.1) +
labs(title= "G nucleotide preceding the edit: JAK1 Exonic gRNAs", x = NULL, y = "z-score: control vs IFNG") +
scale_x_discrete(labels=c("FALSE" = "other base", "TRUE" = "G")) +
annotate("text", x = 1.5, y = 11, label = c("P= 0.0206")) +
annotate("segment", x = 1.1, xend = 1.9, y = 10, yend = 10, alpha = 0.7) +
theme(axis.text.x = element_text(size = rel(1.2))) +
theme_classic()
plot1

plot2 <- ggplot(subset(data_G, sgRNA_type == "stop_essential"), aes(y=zscore_T0_Average, x= G_preceding_edit)) + 
geom_boxplot(colour = "Salmon") +
geom_jitter(alpha = 0.1) +
labs(title= "G nucleotide preceding the edit: stop-essential gRNAs") +
theme_classic()
plot2

###
mean_exonic_G_TRUE <- data_G %>% 
filter(sgRNA_type == "exonic", G_preceding_edit == TRUE) %>%
select(zscore_IFNG_Proliferation_Average) %>%
as.data.frame(mean_exonic_G_TRUE)

mean_exonic_G_FALSE <- data_G %>% 
filter(sgRNA_type == "exonic", G_preceding_edit == FALSE) %>%
select(zscore_IFNG_Proliferation_Average) %>%
as.data.frame(mean_exonic_G_FALSE)

t.test(mean_exonic_G_TRUE, mean_exonic_G_FALSE, var.equal = TRUE)

###
mean_stop_essential_G_TRUE <- data_G %>% 
filter(sgRNA_type == "stop_essential", G_preceding_edit == TRUE) %>%
select(zscore_T0_Average) %>%
as.data.frame(mean_stop_essential_G_TRUE)

mean_stop_essential_G_FALSE <- data_G %>% 
filter(sgRNA_type == "stop_essential", G_preceding_edit == FALSE) %>%
select(zscore_T0_Average) %>%
as.data.frame(mean_stop_essential_G_FALSE)

t.test(mean_stop_essential_G_TRUE, mean_stop_essential_G_FALSE, var.equal = TRUE)
```

```{r}
# find positions of where edit is preceded by a T
data <- mutate(data, T_preceding_edit = ifelse(str_detect(mutant, "Tt") | str_detect(mutant, "aA") == TRUE, TRUE, FALSE))
#filter data for only guides that edit
data_T <- filter(data, window !=edit)
```

```{r}
# T_preceding plot
plot1 <- ggplot(subset(data_T, sgRNA_type == "exonic"), aes(y=zscore_IFNG_Proliferation_Average, x= T_preceding_edit)) + 
geom_boxplot(colour = "Salmon") +
geom_jitter(alpha = 0.1) +
labs(title= "T nucleotide preceding the edit: JAK1 Exonic gRNAs", x = NULL, y = "z-score: control vs IFNG") +
scale_x_discrete(labels=c("FALSE" = "other base", "TRUE" = "T")) +
annotate("text", x = 1.5, y = 11, label = c("P= 0.0003414")) +
annotate("segment", x = 1.1, xend = 1.9, y = 10, yend = 10, alpha = 0.7) +
theme(axis.text.x = element_text(size = rel(1.2))) +
theme_classic()
plot1

plot2 <- ggplot(subset(data_T, sgRNA_type == "stop_essential"), aes(y=zscore_T0_Average, x= T_preceding_edit)) + 
geom_boxplot(colour = "Salmon") +
geom_jitter(alpha = 0.1) +
labs(title= "T nucleotide preceding the edit: Stop Essential vs T0") +
#annotate("text", x = 1.5, y = 2.1, label = c("P= 0.000203")) +
#annotate("segment", x = 1.1, xend = 1.9, y = 1.9, yend = 1.9, alpha = 0.7) +
#theme(axis.text.x = element_text(size = rel(1.1))) +
theme_classic()
plot2

###
mean_exonic_T_TRUE <- data_T %>% 
filter(sgRNA_type == "exonic", T_preceding_edit == TRUE) %>%
select(zscore_IFNG_Proliferation_Average) %>%
as.data.frame(mean_exonic_T_TRUE)

mean_exonic_T_FALSE <- data_T %>% 
filter(sgRNA_type == "exonic", T_preceding_edit == FALSE) %>%
select(zscore_IFNG_Proliferation_Average) %>%
as.data.frame(mean_exonic_T_FALSE)

t.test(mean_exonic_T_TRUE, mean_exonic_T_FALSE, var.equal = TRUE)

###
mean_stop_essential_T_TRUE <- data_T %>% 
filter(sgRNA_type == "stop_essential", T_preceding_edit == TRUE) %>%
select(zscore_T0_Average) %>%
as.data.frame(mean_stop_essential_T_TRUE)

mean_stop_essential_T_FALSE <- data_T %>% 
filter(sgRNA_type == "stop_essential", T_preceding_edit == FALSE) %>%
select(zscore_T0_Average) %>%
as.data.frame(mean_stop_essential_T_FALSE)

t.test(mean_stop_essential_T_TRUE, mean_stop_essential_T_FALSE, var.equal = TRUE)
```

```{r}
# off-target analysis 
data <-  data %>% separate(off_target_summary, remove = FALSE, into = c("zero_mismatches", "one_mismatch", "two_mismatches", "three_mismatches", "four_mismatches"), sep = c(", "))

data<- mutate(data, zero_mismatches = substr(zero_mismatches, 5, end)) %>% mutate(zero_mismatches = as.numeric(zero_mismatches))
data<- mutate(data, one_mismatch = substr(one_mismatch, 4, end)) %>% mutate(one_mismatch = as.numeric(one_mismatch))
data<- mutate(data, two_mismatches = substr(two_mismatches, 4, end)) %>% mutate(two_mismatches = as.numeric(two_mismatches))
data<- mutate(data, three_mismatches = substr(three_mismatches, 4, end)) %>% mutate(three_mismatches = as.numeric(three_mismatches))
data<- mutate(data, four_mismatches = substr(four_mismatches, 4, end))
data<- mutate(data, four_mismatches = str_remove_all(four_mismatches, "\\}")) %>% mutate(four_mismatches = as.numeric(four_mismatches))
```

```{r}
# off_targets
plot1 <- ggplot(subset(data, sgRNA_type == "exonic"), aes(y=zscore_IFNG_Proliferation_Average, x= zero_mismatches)) + 
geom_point(alpha = 0.4, colour = "Salmon") +
labs(title= "gRNA off-target effects: JAK1 Exonic gRNAs \n0 mismatches", y = "z-score: control vs IFNG") +
theme_classic()
plot1

plot2 <- ggplot(subset(data, sgRNA_type == "exonic"), aes(y=zscore_IFNG_Proliferation_Average, x= one_mismatch)) + 
geom_point(alpha = 0.4, colour = "Salmon") +
labs(title= "gRNA off-target effects: JAK1 Exonic gRNAs \n1 mismatch", y = "z-score: control vs IFNG") +
theme_classic()
plot2

plot3 <- ggplot(subset(data, sgRNA_type == "exonic"), aes(y=zscore_IFNG_Proliferation_Average, x= two_mismatches)) + 
geom_point(alpha = 0.4, colour = "Salmon") +
labs(title= "gRNA off-target effects: JAK1 Exonic gRNAs \n2 mismatches", y = "z-score: control vs IFNG") +
theme_classic()
plot3

plot4 <- ggplot(subset(data, sgRNA_type == "exonic"), aes(y=zscore_IFNG_Proliferation_Average, x= three_mismatches)) + 
geom_point(alpha = 0.4, colour = "Salmon") +
labs(title= "gRNA off-target effects: JAK1 Exonic gRNAs \n3 mismatches", y = "z-score: control vs IFNG") +
theme_classic()
plot4

plot5 <- ggplot(subset(data, sgRNA_type == "exonic"), aes(y=zscore_IFNG_Proliferation_Average, x= four_mismatches)) + 
geom_point(alpha = 0.4, colour = "Salmon") +
labs(title= "gRNA off-target effects: JAK1 Exonic gRNAs \n4 mismatches", y = "z-score: control vs IFNG") +
theme_classic()
plot5
```

```{r}
# find positions of edits within the window
data <-  mutate(data, VEP = ifelse(strand == "+", str_locate_all(mutant, pattern = "t"), str_locate_all(mutant, pattern = "a" )))

# index the edits and clean up the syntax for numeric input only
data <-  separate(data, VEP, remove = FALSE, into = c("first", "second", "third", "fourth", "fifth", "sixth", "extra"), sep = c("\\,"), extra = "merge")
data <-  mutate(data, first =(str_replace(first, "integer", "")))
data <-  mutate(data, first =(str_replace(first, "c", "")))
data <-  mutate(data, first =(str_replace(first, "\\(", "")))
data <-  mutate(data, first =(str_replace(first, "\\)", "")))
data <-  mutate(data, second =(str_replace(second, "\\)", "")))
data <-  mutate(data, third =(str_replace(third, "\\)", "")))
data <-  mutate(data, fourth =(str_replace(fourth, "\\)", "")))
data <-  mutate(data, fifth =(str_replace(fifth, "\\)", "")))
data <-  mutate(data, sixth =(str_replace(sixth, "\\)", "")))
```

```{r}
# make positions numeric to determine location from start
data <-  mutate(data, first = as.numeric(first))
data <-  mutate(data, second = as.numeric(second))
data <-  mutate(data, third = as.numeric(third))
data <-  mutate(data, fourth = as.numeric(fourth))
data <-  mutate(data, fifth = as.numeric(fifth))
data <-  mutate(data, sixth = as.numeric(sixth))
data <-  mutate(data, start = as.numeric(start))

# generate single VEP id window mutants 
data <- mutate(data, mut1 = ifelse(strand == "+", paste(chr, (start + (first-1)), (start + (first-1)), "C/T", strand), paste(chr, (start+(first-1)), (start+(first-1)), "G/A", "+")))
data <- mutate(data, mut1 = ifelse(first == "0" | first == "NA", "NA", mut1))

data <- mutate(data, mut2 = ifelse(strand == "+", paste(chr, (start + (second-1)), (start + (second-1)), "C/T", strand), paste(chr, (start+(second-1)), (start+(second-1)), "G/A", "+")))
data <- mutate(data, mut2 = ifelse(second == "NA" | second == first, "NA", mut2))

data <- mutate(data, mut3 = ifelse(strand == "+", paste(chr, (start + (third-1)), (start + (third-1)), "C/T", strand), paste(chr, (start+ (third-1)), (start+ (third-1)), "G/A", "+")))
data <- mutate(data, mut3 = ifelse(third == "NA" | third == first | third == second, "NA", mut3))

data <- mutate(data, mut4 = ifelse(strand == "+", paste(chr, (start + (fourth-1)), (start + (fourth-1)), "C/T", strand), paste(chr, (start+ (fourth-1)), (start+ (fourth-1)), "G/A", "+")))
data <- mutate(data, mut4 = ifelse(fourth == "NA" | fourth == first | fourth == second | fourth == third, "NA", mut4))

data <- mutate(data, mut5 = ifelse(strand == "+", paste(chr, (start + (fifth-1)), (start + (fifth-1)), "C/T", strand), paste(chr, (start+ (fifth-1)), (start + (fifth-1)), "G/A", "+")))
data <- mutate(data, mut5 = ifelse(fifth == "NA" | fifth == first | fifth == second | fifth == third | fifth == fourth, "NA", mut5))

data <- mutate(data, mut6 = ifelse(strand == "+", paste(chr, (start + (sixth-1)), (start + (sixth-1)), "C/T", strand), paste(chr, (start+ (sixth-1)), (start + (sixth-1)), "G/A", "+")))
data <- mutate(data, mut6 = ifelse(sixth == "NA" | sixth == first | sixth == second | sixth == third | sixth == fourth | sixth == fifth, "NA", mut6))

# generate combined VEP id window mutants
data <- mutate(data, combined_mut = ifelse(window != edit & strand == "+", paste0(chr, " ", (start+3), " ",(start+8), " ", window, "/", str_to_upper(edit), " ", "+"), "NA"))
data <- mutate(data, combined_mut = ifelse(window != edit & strand == "-", paste0(chr, " ", (start+14), " ",(start+19), " ", window, "/", str_to_upper(edit), " ", "+"), combined_mut))

# melt data and eliminate NA in VEP_id
data <- melt(select(data, !c(VEP, extra)), measure.vars = c("mut1", "mut2", "mut3", "mut4", "mut5", "mut6", "combined_mut"), variable.name = "edit_index", value.name = "VEP_id")
data <- mutate(data, VEP_id = str_replace_all(string = VEP_id, pattern  = "NA", replacement = ""))
data <- mutate(data, VEP_id = str_replace_na(string = VEP_id, replacement = ""))

# generate tsv output file
data %>% filter(Gene == "JAK1") %>% arrange(start) %>% select("VEP_id") %>% write_tsv("VEP_query.tsv")

# Pass VEP_query.tsv to VEP ensembl and include only MANE results using online filter
# Make a results file called VEP_results.csv
# Change excel cells to numbers as it gives dates automatically which messes things up
# Remove duplicate values for VEP_id entries before joining!
# Get rid of unwanted data output columns in excel
# you can replace commas in text editor beofre making hte csv as this can include additional delimeted columns in excel
```

```{r}
# read in VEP results file
VEP_results <- as_tibble(read.csv("JAK1_VEP_results_010420.csv", stringsAsFactors = FALSE, header = TRUE))
VEP_results <- rename(VEP_results, Uploaded_variation = "X.Uploaded_variation")

data <- separate(data, VEP_id, remove = TRUE, into = c("chr_edit", "start_edit", "end_edit", "edit_genotype", "extra_"), sep = c(" "), extra = "merge")
```

```{r}
# make joining by query result compatible
data <- mutate(data, Uploaded_variation = (paste0(chr_edit, "_", start_edit, "_", edit_genotype)))
VEP_results <- rename(VEP_results, Gene_ID_VEP = Gene)

# join VEP results to gRNA table
data <- left_join(data, VEP_results)
data <- data %>% select(!extra_)
data <- data %>% select(!"first" & !"second" & !"third" & !"fourth" & !"fifth")
```
```{r}
# look for guides which are not represented in the protein sequence
data %>% filter(sgRNA_ID %in% c(
"908510445",
"908509673",
"908509674", 
"908509675", 
"908509006", 
"908511164", 
"908511165", 
"908511166", 
"908512109",
"908523459",	
"908510287",		
"908509235",
"908513051",
"908523399",
"908523616")) %>%
  select(sgRNA_ID, Consequence, edit_index, sgRNA_type, zscore_IFNG_Proliferation_Average) %>%
  distinct() %>%
    group_by(sgRNA_ID)
```

```{r}
# split HGVSp output from VEP to give Amino_Acid_Position
data <- separate(data, HGVSp, remove = FALSE, into = c("HGVSp1", "HGVSp2"), sep = c(":"), extra = "merge")

data <- mutate(data, Amino_Acid_Position = substr(HGVSp2, 6, 50))
data <- mutate(data, Amino_Acid_Position = ifelse(str_detect(Amino_Acid_Position, "_"), "multiple edit", Amino_Acid_Position))
data <- mutate(data, Amino_Acid_Position = ifelse(str_detect(Amino_Acid_Position, "%"), "synonymous", Amino_Acid_Position))
data <- mutate(data, Amino_Acid_Position = ifelse(Amino_Acid_Position != "multiple edit" & Amino_Acid_Position != "synonymous", str_sub(Amino_Acid_Position, end = -4), Amino_Acid_Position))
```

```{r}
# query COSMIC for the gene and downlaod the file
# save it as COSMIC_results.csv
# read in COSMIC results file
COSMIC_results <- as_tibble(read.csv("JAK1_test_COSMIC_results.csv", stringsAsFactors = FALSE, header = TRUE))
COSMIC_results <- rename(COSMIC_results, Amino_Acid_Position = "Position")
COSMIC_results <- mutate(COSMIC_results, Amino_Acid_Position = (as.character(Amino_Acid_Position)))

# filter out unknown and synonymous COSMIC variants
COSMIC_results <- filter(COSMIC_results, Amino_Acid_Position != "" & Type != "Unkown" & Type != "Substitution - coding silent")

# join COSMIC_results by Amino_Acid_Position
data <- left_join(data, COSMIC_results)
data <- as_tibble(data)

# make COSMIC AA.Mutations into three letter code for compatibility
data <- mutate(data, AA.Mutation_Cosmic = str_replace_all(string = AA.Mutation, 
                c("A" = "Ala",
                  "R" = "Arg",
                  "N" = "Asn",
                  "D" = "Asp",
                  "C" = "Cys",
                  "G" = "Gly",
                  "E" = "Glu",
                  "Q" = "Gln",
                  "H" = "His",
                  "I" = "Ile",
                  "L" = "Leu",
                  "K" = "Lys",
                  "M" = "Met",
                  "P" = "Pro",
                  "F" = "Phe",
                  "S" = "Ser",
                  "T" = "Thr",
                  "W" = "Trp",
                  "Y" = "Tyr",
                  "V" = "Val",
                  "\\*" = "Ter"))
               )
# alter names 
data <- rename(data, Amino_Acid_Change = HGVSp2)
data <- rename(data, COSMIC_Count = Count)
```

```{r}
# JAK1 protein. This has rows with number of amino acid and synonymous and mulitple_edit additional rows to include these VEP variants
JAK1_protein <- as_tibble(read.csv("Protein_length.csv", stringsAsFactors = FALSE, header = TRUE))
JAK1_protein <- mutate(JAK1_protein, Amino_Acid_Position = as.character(Amino_Acid_Position))
data <- left_join(JAK1_protein, data)
```

```{r}
# JAK1 phosphositeplus data
JAK1_phosphosites <- as_tibble(read.csv("PhosphoSitePlus_JAK1.csv", stringsAsFactors = FALSE, header = TRUE))
JAK1_phosphosites <- mutate(JAK1_phosphosites, Amino_Acid_Position = as.character(Amino_Acid_Position))
data <- left_join(data, JAK1_phosphosites)
```

```{r}
# define codon change
substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

data <- mutate(data, codon  = str_sub(Amino_Acid_Change,3,5))
data <- mutate(data, new_codon  = substrRight(Amino_Acid_Change, 3))
data <- mutate(data, new_codon = ifelse(new_codon == "%3D", codon, new_codon))
data <- mutate(data, Codon_Change = paste0(codon, ">", new_codon)) 
```

```{r}
# COSMIC_Count graph
data <- mutate(data, COSMIC_Count = ifelse(is.na(COSMIC_Count) == TRUE, 0, COSMIC_Count))
data_COSMIC_graph <- data %>%
  filter(Gene == "JAK1" & Amino_Acid_Position != "synonymous" & Amino_Acid_Position != "multiple edit") %>%
  select(COSMIC_Count, Amino_Acid_Position, sgRNA_ID, zscore_IFNG_Proliferation_Average, AA.Mutation_Cosmic) %>% 
  distinct()
data_COSMIC_graph
```

```{r}
# COSMIC_Count graph
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(data_COSMIC_graph, aes(x=Amino_Acid_Position, y= zscore_IFNG_Proliferation_Average, colour = COSMIC_Count)) +
geom_jitter(size=3, alpha=0.7, position = pos) +
labs(title= "JAK1 protein gRNAs", x = "amino acid position", y = "z-score: proliferation screen") +
theme_classic()
plot1

plot2 <- ggplot(data_COSMIC_graph, aes(x=COSMIC_Count, y= zscore_IFNG_Proliferation_Average)) +
geom_jitter(size=3, alpha=0.7, position = pos, colour = ifelse(data_COSMIC_graph$zscore_IFNG_Proliferation_Average >5 & data_COSMIC_graph$COSMIC_Count >0, "royalblue1", ifelse(data_COSMIC_graph$zscore_IFNG_Proliferation_Average < -2.5 & data_COSMIC_graph$COSMIC_Count >0, "salmon","grey45"))) +
labs(title= "JAK1 gRNAs, protein coding", x = "COSMIC Count", y = "z-score: proliferation screen") +
geom_hline(yintercept=c(5,5), linetype="dotted", colour = "royalblue1") +
geom_hline(yintercept=c(0,0), linetype="dotted", colour = "grey45") +
geom_hline(yintercept=c(-2.5,-2.5), linetype="dotted", colour = "salmon") +
geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_Average >5 & COSMIC_Count > 0 | zscore_IFNG_Proliferation_Average < -2.5 & COSMIC_Count > 0, AA.Mutation_Cosmic, "")), size = 2, position = pos) +
scale_x_continuous(breaks = c(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20)) +
theme_classic()
plot2
```


```{r}
# read in Master_JAK1_Protein_Mutations_Humans
Mutations <- as_tibble(read.csv("Master_JAK1_Protein_Mutations_Humans.csv", stringsAsFactors = FALSE, header = TRUE))
Mutations <- mutate(Mutations, Amino_Acid_Position = as.numeric(Amino_Acid_Position))
data <- mutate(data, Amino_Acid_Position = as.numeric(Amino_Acid_Position))
data <- left_join(data, Mutations)

# make COSMIC AA.Mutations into three letter code for compatibility
data <- mutate(data, Clinical_Protein_Change_ThreeL = ifelse(Citation != "gnomAD_v3.1", str_replace_all(string = Clinical_Protein_Change, 
                c("A" = "Ala",
                  "R" = "Arg",
                  "N" = "Asn",
                  "D" = "Asp",
                  "C" = "Cys",
                  "G" = "Gly",
                  "E" = "Glu",
                  "Q" = "Gln",
                  "H" = "His",
                  "I" = "Ile",
                  "L" = "Leu",
                  "K" = "Lys",
                  "M" = "Met",
                  "P" = "Pro",
                  "F" = "Phe",
                  "S" = "Ser",
                  "T" = "Thr",
                  "W" = "Trp",
                  "Y" = "Tyr",
                  "V" = "Val",
                  "\\*" = "Ter")), Clinical_Protein_Change))

data <- select(data, !Clinical_Protein_Change)
data <- rename(data, Clinical_Protein_Change = Clinical_Protein_Change_ThreeL)
```

```{r}
# all_hits_zscores_protein
all_hits_zscores_protein <- filter(data, zscore_IFNG_Proliferation_Average > 5  | zscore_IFNG_Proliferation_Average < -2.5 | zscore_IFNG_FACS_Average > 5) %>%
  filter(Gene == "JAK1") %>%
  distinct()
```

```{r}
# compare hits from all guides and just protein
all_hits_zscores <- select(all_hits_zscores, sgRNA_ID, zscore_IFNG_Proliferation_Average) %>% distinct()
all_hits_zscores_protein <- select(all_hits_zscores_protein, sgRNA_ID, zscore_IFNG_Proliferation_Average) %>% distinct()

missing_from_protein <- anti_join(all_hits_zscores, all_hits_zscores_protein)

all_hits_zscores
all_hits_zscores_protein
missing_from_protein
```

```{r}
# write final output.csv file without filtering
write_csv(data, "final.csv")
```

```{r}
# find_anomalies between FACS and Proliferation
data %>% filter(zscore_IFNG_FACS_Average > 5 & zscore_IFNG_Proliferation_Average <5) %>% select(sgRNA_ID, Amino_Acid_Position, zscore_IFNG_Proliferation_Average, zscore_IFNG_FACS_Average) %>% distinct()
data %>% filter(zscore_IFNG_Proliferation_Average > 5 & zscore_IFNG_FACS_Average <5) %>% select(sgRNA_ID, Amino_Acid_Position, zscore_IFNG_Proliferation_Average, zscore_IFNG_FACS_Average) %>% distinct()
```

```{r}
#graphical subset to reduce Amino_Acid_Position redundant duplicate plotting
data_graphing <- data %>%
  select(Amino_Acid_Position, Gene, UniprotDomain, Protein_Feature, sgRNA_ID, sgRNA_type, Consequence, chr, start, end, seq, PTM, Amino_Acid_Change, Clinical_Protein_Change, Citation, L2FC_Proliferation_1,	L2FC_Proliferation_2, L2FC_FACS_1,	L2FC_FACS_2, L2FC_T0_1, L2FC_T0_2, L2FC_Plasmid_1, L2FC_Plasmid_2, Average_L2FC_Proliferation, Average_L2FC_FACS, off_target_summary, zscore_IFNG_Proliferation_Average, zscore_IFNG_FACS_Average, zscore_IFNG_Proliferation_1, zscore_IFNG_FACS_1, zscore_IFNG_Proliferation_2, zscore_IFNG_FACS_2) %>%
  distinct()
```

```{r}
#filter data_graphing subset to get rid of duplicate values and summarise them into one row per Amino_Acid_Position
Amino_Acid_Change <- data_graphing %>%
  group_by(Amino_Acid_Position) %>%
  distinct(Amino_Acid_Change) %>%
  summarise(Amino_Acid_Change = paste(Amino_Acid_Change, collapse = ", "))

Clinical_Protein_Change <- data_graphing %>%
  group_by(Amino_Acid_Position) %>%
  distinct(Clinical_Protein_Change) %>%
  summarise(Clinical_Protein_Change = paste(Clinical_Protein_Change, collapse = ", "))

Citation <- data_graphing %>%
  group_by(Amino_Acid_Position) %>%
  distinct(Citation) %>%
  summarise(Citation = paste(Citation, collapse = ", "))

PTM <- data_graphing %>%
  group_by(Amino_Acid_Position) %>%
  distinct(PTM) %>%
  summarise(PTM = paste(PTM, collapse = ", "))

data_graphing_filtered <- select(data_graphing, !c(Amino_Acid_Change, Clinical_Protein_Change, Citation, PTM))

data_graphing_filtered <- left_join(data_graphing_filtered, Amino_Acid_Change) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, Clinical_Protein_Change) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, Citation) %>% distinct()
data_graphing_filtered <- left_join(data_graphing_filtered, PTM) %>% distinct()

data_graphing_filtered
```

```{r}
# dot plots
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, Gene =="JAK1" & Consequence != "synonymous_variant"), aes(x=Amino_Acid_Position, y= zscore_IFNG_Proliferation_Average, colour = Consequence)) + 
geom_point(size=3, alpha=0.85, position = pos, shape = 16) +
geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_Average >5 & PTM != "NA" | zscore_IFNG_Proliferation_Average < -2.5 & PTM != "NA", PTM, "")), size = 3, position = pos) +
geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_Average >5 & Protein_Feature != "NA" | zscore_IFNG_Proliferation_Average < -2.5 & Protein_Feature != "NA", Protein_Feature, "")), size = 3, position = pos) +
geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_Average >5  & Clinical_Protein_Change != "NA" | zscore_IFNG_Proliferation_Average < -2.5 & Clinical_Protein_Change != "NA", paste0(Amino_Acid_Change, "\n", "(", Clinical_Protein_Change, " - ", Citation, ")"), "")), size = 3, position = pos) +
labs(title= "BE edit prediction (C>T): JAK1 - proliferation screen", x="amino acid position", y = "z-score: control vs IFNG") +
scale_colour_manual(values=c("grey45", "royalblue1", "royalblue1","salmon", "salmon")) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_x_continuous(breaks = c(0, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1100, 1200)) +
theme_classic()
plot1

plot2 <- ggplot(subset(data_graphing_filtered, Gene =="JAK1" & Consequence != "synonymous_variant"), aes(x=Amino_Acid_Position, y= zscore_IFNG_FACS_Average, colour = Consequence)) + 
geom_point(size=3, alpha=0.85, position = pos, shape = 16) +
geom_text_repel(aes(label=ifelse(zscore_IFNG_FACS_Average >5 & PTM != "NA" | zscore_IFNG_FACS_Average < -2.5 & PTM != "NA", PTM, "")), size = 3, position = pos) +
geom_text_repel(aes(label=ifelse(zscore_IFNG_FACS_Average >5 & Protein_Feature != "NA" | zscore_IFNG_FACS_Average < -2.5 & Protein_Feature != "NA", Protein_Feature, "")),  size = 3, position = pos) +
geom_text_repel(aes(label=ifelse(zscore_IFNG_FACS_Average >5  & Clinical_Protein_Change != "NA" | zscore_IFNG_FACS_Average < -2.5 & Clinical_Protein_Change != "NA", paste0(Amino_Acid_Change, "\n", "(", Clinical_Protein_Change, " - ", Citation, ")"), "")), size = 3, position = pos) +
labs(title= "BE edit prediction (C>T): JAK1 - FACS screen", x="amino acid position", y = "z-score: control vs IFNG") +
scale_colour_manual(values=c("grey45", "royalblue1", "royalblue1","salmon", "salmon")) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_x_continuous(breaks = c(0, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1100, 1200)) +
theme_classic()
plot2
```

```{r}
# correlation plots
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(data_graphing_filtered, Gene =="JAK1"), aes(x=zscore_IFNG_Proliferation_1, y= zscore_IFNG_FACS_1)) + 
geom_jitter(size=3, alpha = 0.7, position = pos) +
labs(title= "BE edit prediction (C>T): JAK1") +
geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_1 >5 & zscore_IFNG_FACS_1 > 5 & Clinical_Protein_Change != "NA", paste0(Amino_Acid_Change, "\n", "(", Clinical_Protein_Change, " - ", Citation, ")"), "")), size = 3, position = pos) +
geom_smooth(method = lm) +
theme_classic()
plot1

plot2 <- ggplot(subset(data_graphing_filtered, Gene =="JAK1"), aes(x=zscore_IFNG_Proliferation_2, y= zscore_IFNG_FACS_2)) + 
geom_point(size=3, alpha=0.7, position = pos) +
labs(title= "BE edit prediction (C>T): JAK1") +
geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_2 >5 & zscore_IFNG_FACS_2 > 5 & Clinical_Protein_Change != "NA", paste0(Amino_Acid_Change, "\n", "(", Clinical_Protein_Change, " - ", Citation, ")"), "")), size = 3, position = pos) +
geom_smooth(method = lm) +
theme_classic()
plot2

plot3 <- ggplot(subset(data_graphing_filtered, Gene =="JAK1"), aes(x=zscore_IFNG_Proliferation_Average, y= zscore_IFNG_FACS_Average)) + 
geom_point(size=3, alpha=0.7, position = pos) +
labs(title= "BE edit prediction (C>T): JAK1") +
geom_text_repel(aes(label=ifelse(zscore_IFNG_Proliferation_Average >5 & zscore_IFNG_FACS_Average > 5 & Clinical_Protein_Change != "NA", paste0(Amino_Acid_Change, "\n", "(", Clinical_Protein_Change, " - ", Citation, ")"), "")), size = 3, position = pos) +
geom_smooth(method = lm) +
theme_classic()
plot3
```


```{r}
# check protein_hits zscores
data_graphing_filtered %>% filter(zscore_IFNG_Proliferation_Average >5 & Gene == "JAK1"| zscore_IFNG_Proliferation_Average < -2.5 & Gene == "JAK1"| zscore_IFNG_FACS_Average >5 & Gene == "JAK1") %>% 
  distinct()
```

```{r}
# categorise codons
nonpolar <- c("Phe", "Leu", "Ile", "Val", "Pro", "Ala", "Trp", "Gly")
polar <- c("Ser", "Thr", "Tyr", "Gln", "Asp", "Cys", "Asn")
basic <- c("His", "Lys", "Arg")
acidic <- c("Asp", "Glu")
stop <- c("Ter")
initiation <- c("Met")

data <- mutate(data, codon_class = ifelse(codon %in% nonpolar, "nonpolar", 
                                          ifelse(codon %in% polar, "polar",
                                                 ifelse(codon %in% basic, "basic",
                                                        ifelse(codon %in% acidic, "acidic",
                                                               ifelse(codon %in% stop, "stop",
                                                                      ifelse(codon %in% initiation, "initiation", 
                                                                             "no class")))))))

data <- mutate(data, new_codon_class = ifelse(new_codon %in% nonpolar, "nonpolar", 
                                          ifelse(new_codon %in% polar, "polar",
                                                 ifelse(new_codon %in% basic, "basic",
                                                        ifelse(new_codon %in% acidic, "acidic",
                                                               ifelse(new_codon %in% stop, "stop",
                                                                      ifelse(new_codon %in% initiation, "initiation", 
                                                                             "no class")))))))

data <- mutate(data, codon_class_change = paste0(codon_class, ">", new_codon_class))
data <- mutate(data, codon_class_change = ifelse(new_codon_class == codon_class, "no change", codon_class_change))
```

```{r}
# codon plot
data %>% select(codon, new_codon, codon_class_change)

plot1 <- ggplot(subset(data, is.na(new_codon) == FALSE), aes(x=Codon_Change, fill = codon_class_change)) +
geom_bar() +
labs(title = "Predicted C>T codon transitions", x = "codon change", fill = "chemical properties") +
theme_classic() +
theme(axis.text.x = element_text(angle = 90))
plot1
```

```{r}
# codon plot 2
data %>% select(codon, new_codon, codon_class_change, zscore_IFNG_Proliferation_Average)

plot1 <- ggplot(subset(data, is.na(new_codon) == FALSE & Gene == "JAK1"), aes(x=codon_class_change, y= zscore_IFNG_Proliferation_Average)) +
geom_boxplot(colour = "salmon") +
geom_jitter(alpha = 0.2) +
labs(title = "Predicted C>T codon transitions") +
theme_classic() +
theme(axis.text.x = element_text(angle = 90))
plot1
```

```{r}
# Validation
# read in hits, promoter, 5'UTR, clinical hits with L2FC_Promoter>1 and L2FC_FACS >1
hits <- as_tibble(read.csv("hit_PRIMERS_validation_041220.csv", stringsAsFactors = FALSE, header = TRUE))
hits <- mutate(hits, sgRNA_ID = as.character(sgRNA_ID))
annotation_hits <- data %>% select(sgRNA_ID, L2FC_Proliferation_1, L2FC_Proliferation_2, L2FC_FACS_1, L2FC_FACS_2, Average_L2FC_Proliferation, Average_L2FC_FACS, zscore_IFNG_Proliferation_Average, zscore_IFNG_FACS_Average, G_guide, for_primer, rev_primer, Amino_Acid_Position, Amino_Acid_Change, Clinical_Protein_Change, Citation, PHENOTYPES, COSMIC_Count, UniprotDomain, Protein_Feature) %>% distinct()
hits <- left_join(hits, annotation_hits)
write.csv(hits, "hits_primers_validation.csv")
```

```{r}
# replicate correlation
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(data_graphing_filtered, aes(x=zscore_IFNG_Proliferation_1, y= zscore_IFNG_Proliferation_2, colour = zscore_IFNG_FACS_Average)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "BE edit prediction (C>T): JAK1") +
geom_smooth(method = lm) +
geom_vline(xintercept=c(5,5),linetype="dotted") +
geom_hline(yintercept=c(5,5),linetype="dotted") +
#geom_segment(aes(x= -1, y= -1, xend = -5, yend = -1), colour = "red", linetype="dotted") +
#geom_segment(aes(x= -1, y= -1, xend = -1, yend = -5), colour = "red", linetype="dotted") +
#geom_text_repel(aes(label=ifelse(sgRNA_ID %in% c("908523459", "908523616", "908514537", "908514536", "908509432", "908510622", "908509431", "908510623", "908508937", "908509672", "908509010", "908512440", "908510034", "908510164", "908509007", "908510033", "908510274", "908509428", "908510028", "908510170", "908510032", "908508944", "908508948", "908510162"), Amino_Acid_Position, "")), size = 3, position = pos) +
theme_classic()
plot1

plot2 <- ggplot(data_graphing_filtered, aes(x=zscore_IFNG_FACS_1, y= zscore_IFNG_FACS_2, colour = zscore_IFNG_Proliferation_Average)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "BE edit prediction (C>T): JAK1") +
geom_smooth(method = lm) +
geom_vline(xintercept=c(5,5),linetype="dotted") +
geom_hline(yintercept=c(5,5),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(sgRNA_ID %in% c("908523459", "908523616", "908514537", "908514536", "908509432", "908510622", "908509431", "908510623", "908508937", "908509672", "908509010", "908512440", "908510034", "908510164", "908509007", "908510033", "908510274", "908509428", "908510028", "908510170", "908510032", "908508944", "908508948", "908510162"), Amino_Acid_Position, "")), size = 3, position = pos) +
theme_classic()
plot2
```

```{r}
# zscore and L2FC correlation and cut-offs
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(data_graphing_filtered, aes(x=Average_L2FC_Proliferation, y= zscore_IFNG_Proliferation_Average, colour = Average_L2FC_FACS)) + 
geom_point(size=3, alpha=0.85, position = pos, shape = 16) +
labs(title= "JAK1 protein amino acids") +
geom_vline(xintercept=c(1,1),linetype="dotted") +
geom_hline(yintercept=c(5,5),linetype="dotted") +
geom_vline(xintercept=c(-0.5,-0.5),linetype="dotted", colour = "salmon") +
geom_hline(yintercept=c(-1,-1),linetype="dotted", colour = "salmon") +
geom_text_repel(aes(label=ifelse(sgRNA_ID %in% c("908523459", "908523616", "908514537", "908514536", "908509432", "908510622", "908509431", "908510623", "908508937", "908509672", "908509010", "908512440", "908510034", "908510164", "908509007", "908510033", "908510274", "908509428", "908510028", "908510170", "908510032", "908508944", "908508948", "908510162"), Amino_Acid_Position, "")), size = 3, position = pos) +
theme_classic()
plot1

plot2 <- ggplot(data_graphing_filtered, aes(x=Average_L2FC_FACS, y= zscore_IFNG_FACS_Average, colour = Average_L2FC_Proliferation)) + 
geom_point(size=3, alpha=0.85, position = pos, shape = 16) +
labs(title= "JAK1 protein amino acids") +
geom_vline(xintercept=c(1,1),linetype="dotted") +
geom_hline(yintercept=c(5,5),linetype="dotted") +
geom_text_repel(aes(label=ifelse(sgRNA_ID %in% c("908523459", "908523616", "908514537", "908514536", "908509432", "908510622", "908509431", "908510623", "908508937", "908509672", "908509010", "908512440", "908510034", "908510164", "908509007", "908510033", "908510274", "908509428", "908510028", "908510170", "908510032", "908508944", "908508948", "908510162"), Amino_Acid_Position, "")), size = 3, position = pos) +
theme_classic()
plot2

plot3 <- ggplot(data_graphing_filtered, aes(x=zscore_IFNG_Proliferation_Average, y= zscore_IFNG_FACS_Average)) + 
geom_point(size=3, alpha=0.7, position = pos, shape = 16, colour = ifelse(data_graphing_filtered$zscore_IFNG_Proliferation_Average < -2.5, "salmon", ifelse(data_graphing_filtered$zscore_IFNG_Proliferation_Average > 5 & data_graphing_filtered$zscore_IFNG_FACS_Average > 5, "royalblue1", "grey45"))) +
labs(title= "JAK1 protein amino acids", x = "z-score: proliferation screen", y = "z-score: FACS screen") +
geom_vline(xintercept=c(5,5),linetype="dotted") +
geom_hline(yintercept=c(5,5),linetype="dotted") +
geom_vline(xintercept=c(-2.5,-2.5),linetype="dotted", colour = "salmon") +
geom_text_repel(aes(label=ifelse(sgRNA_ID %in% c("908523459", "908523616", "908514537", "908514536", "908509432", "908510622", "908509431", "908510623", "908508937", "908509672", "908509010", "908512440", "908510034", "908510164", "908509007", "908510033", "908510274", "908509428", "908510028", "908510170", "908510032", "908508944", "908508948", "908510162"), Amino_Acid_Position, "")), size = 3, position = pos) +
theme_classic()
plot3
```

```{r}
# SIFT, PolyPhen, BLOSUM62, IMPACT, Consequence, LoFtool, correlation
data_prediction_tools <- data %>% select(sgRNA_ID, Amino_Acid_Change, zscore_IFNG_Proliferation_Average, SIFT, PolyPhen, BLOSUM62, IMPACT, Consequence, LoFtool) %>% 
  distinct()

data_prediction_tools <- mutate(data_prediction_tools, SIFT = as.numeric(SIFT))
data_prediction_tools <- mutate(data_prediction_tools, PolyPhen = as.numeric(PolyPhen))
data_prediction_tools <- mutate(data_prediction_tools, BLOSUM62 = as.numeric(BLOSUM62))
data_prediction_tools <- mutate(data_prediction_tools, LoFtool = as.numeric(LoFtool))
data_prediction_tools

#Consequence
plot1 <- ggplot(subset(data_prediction_tools, Consequence != "NA"), aes(x=zscore_IFNG_Proliferation_Average, y= Consequence)) + 
geom_violin(colour = "salmon") +
geom_jitter(size=3, alpha=0.2, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_bw()
plot1


# SIFT 0 is deleterious, 1 is tolerated
plot2 <- ggplot(data_prediction_tools, aes(x=zscore_IFNG_Proliferation_Average, y= SIFT)) +
geom_jitter(size=3, alpha=0.2, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_bw()
plot2

#PolyPhen 0 is benign, 1 is damaging
plot3 <- ggplot(data_prediction_tools, aes(x=zscore_IFNG_Proliferation_Average, y= PolyPhen)) +
geom_jitter(size=3, alpha=0.2, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_bw()
plot3

#BLOSUM62 positive is conserved, negative is not conserved
plot4 <- ggplot(data_prediction_tools, aes(x=zscore_IFNG_Proliferation_Average, y= BLOSUM62)) +
geom_jitter(size=3, alpha=0.2, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_bw()
plot4

#IMPACT High Low Intermediate
plot5 <- ggplot(subset(data_prediction_tools, Consequence != "NA"), aes(x=zscore_IFNG_Proliferation_Average, y= IMPACT)) +
geom_violin(colour = "salmon") +
geom_jitter(size=3, alpha=0.2, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_bw()
plot5

#LoFTool
plot6 <- ggplot(data_prediction_tools, aes(x=zscore_IFNG_Proliferation_Average, y= LoFtool)) +
geom_jitter(size=3, alpha=0.2, width = 0.2, height = 0.2) +
labs(title= "BE edit prediction (C>T): JAK1") +
theme_bw()
plot6
```

```{r}
# PyMol
data %>% 
  arrange(Amino_Acid_Position) %>%
  group_by(Amino_Acid_Position) %>%
  mutate(Average_L2FC_Proliferation_unique = mean(Average_L2FC_Proliferation)) %>%
  select(Amino_Acid_Position, Average_L2FC_Proliferation_unique) %>% 
  distinct() %>%
  write.csv("newBfactors.csv")
```

```{r}
# Check for coverage of amino acids
data_graphing_filtered %>% filter(is.na(Amino_Acid_Position) == FALSE) %>% 
  select(Amino_Acid_Position, sgRNA_type) %>%
    distinct() %>%
      count()

data_graphing_filtered %>% filter(is.na(Amino_Acid_Position) == FALSE) %>% 
  select(Amino_Acid_Position, sgRNA_type) %>%
    distinct() %>%
      filter(is.na(sgRNA_type) == FALSE) %>%
        count()

data_graphing_filtered %>% filter(is.na(Amino_Acid_Position) == FALSE) %>% 
  select(Amino_Acid_Position, sgRNA_type) %>%
    distinct() %>%
      filter(is.na(sgRNA_type)) %>%
        count()

#percentage amino acid coverage of JAK1 with NGG BE3
239/1154*100
```
